{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Thanks to Kriti Mahajan. This table was developed with study type features extracted from the meta data.\n",
    "Code found here: https://colab.research.google.com/drive/1lay-yWJ5ohN94lQLHYIxtWMuDWtCRGy0#scrollTo=N-IhAeEkcxDe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>paper_id</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>url</th>\n",
       "      <th>Statistical Methods</th>\n",
       "      <th>statistical_method_details</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>study_type_details</th>\n",
       "      <th>results</th>\n",
       "      <th>coronavirus</th>\n",
       "      <th>fatality</th>\n",
       "      <th>sample_size</th>\n",
       "      <th>paper_focus</th>\n",
       "      <th>variables_controlled_for</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>Sequence requirements for RNA strand transfer ...</td>\n",
       "      <td>Nidovirus subgenomic mRNAs contain a leader se...</td>\n",
       "      <td>2001-12-17</td>\n",
       "      <td>Pasternak, Alexander O.; van den Born, Erwin; ...</td>\n",
       "      <td>http://europepmc.org/articles/pmc125340?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['genom', 'mutagenesis']</td>\n",
       "      <td>Nidovirus subgenomic mRNAs contain a leader se...</td>\n",
       "      <td>Here we present results of a comprehensive co-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>Crystal structure of murine sCEACAM1a[1,4]: a ...</td>\n",
       "      <td>CEACAM1 is a member of the carcinoembryonic an...</td>\n",
       "      <td>2002-05-01</td>\n",
       "      <td>Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...</td>\n",
       "      <td>http://europepmc.org/articles/pmc125375?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['protein']</td>\n",
       "      <td>We discuss the structural basis of virus recep...</td>\n",
       "      <td>Here we report the crystal structure of solubl...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>Synthesis of a novel hepatitis C virus protein...</td>\n",
       "      <td>Hepatitis C virus (HCV) is an important human ...</td>\n",
       "      <td>2001-07-16</td>\n",
       "      <td>Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['protein', 'genom']</td>\n",
       "      <td>Its RNA genome codes for a polyprotein, which ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>Structure of coronavirus main proteinase revea...</td>\n",
       "      <td>The key enzyme in coronavirus polyprotein proc...</td>\n",
       "      <td>2002-07-01</td>\n",
       "      <td>Anand, Kanchan; Palm, Gottfried J.; Mesters, J...</td>\n",
       "      <td>http://europepmc.org/articles/pmc126080?pdf=re...</td>\n",
       "      <td>['model']</td>\n",
       "      <td>Molecular modelling and mutagenesis data impli...</td>\n",
       "      <td>['protein', 'mutagenesis']</td>\n",
       "      <td>The key enzyme in coronavirus polyprotein proc...</td>\n",
       "      <td>The study illustrates that RNA viruses have ev...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>Discontinuous and non-discontinuous subgenomic...</td>\n",
       "      <td>Arteri-, corona-, toro- and roniviruses are ev...</td>\n",
       "      <td>2002-12-01</td>\n",
       "      <td>van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...</td>\n",
       "      <td>http://europepmc.org/articles/pmc136939?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['genom']</td>\n",
       "      <td>During infection, multiple subgenomic (sg) mRN...</td>\n",
       "      <td>We show that BEV mRNAs 3–5 lack a leader seque...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56640</th>\n",
       "      <td>56640</td>\n",
       "      <td>rn4gm3sc</td>\n",
       "      <td>Structure-Based Stabilization of Non-native Pr...</td>\n",
       "      <td>[Image: see text] Structure-based stabilizatio...</td>\n",
       "      <td>2020-02-27</td>\n",
       "      <td>Lin, Shan-Meng; Lin, Shih-Chao; Hsu, Jia-Ning;...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['protein']</td>\n",
       "      <td>[Image: see text] Structure-based stabilizatio...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Here, we identified a non-native interaction i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56641</th>\n",
       "      <td>56641</td>\n",
       "      <td>l5oecoot</td>\n",
       "      <td>Immunomodulators as an antimicrobial tool</td>\n",
       "      <td>The spectrum of infectious diseases has shifte...</td>\n",
       "      <td>2006-10-31</td>\n",
       "      <td>Pirofski, Liise-anne; Casadevall, Arturo</td>\n",
       "      <td>https://doi.org/10.1016/j.mib.2006.08.004</td>\n",
       "      <td>[' tool']</td>\n",
       "      <td>However, to date only a few immunomodulators h...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56642</th>\n",
       "      <td>56642</td>\n",
       "      <td>i4acg2ex</td>\n",
       "      <td>Discourses of disease: Representations of tube...</td>\n",
       "      <td>Abstract This paper critically examines the wa...</td>\n",
       "      <td>2008-02-29</td>\n",
       "      <td>Lawrence, Jody; Kearns, Robin A.; Park, Julie;...</td>\n",
       "      <td>https://doi.org/10.1016/j.socscimed.2007.10.015</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56643</th>\n",
       "      <td>56643</td>\n",
       "      <td>rve0w3dg</td>\n",
       "      <td>Equine Infectious Anemia</td>\n",
       "      <td>This article describes clinical signs and tran...</td>\n",
       "      <td>1993-08-31</td>\n",
       "      <td>Sellon, Debra Clabough</td>\n",
       "      <td>https://doi.org/10.1016/s0749-0739(17)30399-1</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['review', 'clinical']</td>\n",
       "      <td>This article describes clinical signs and tran...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56644</th>\n",
       "      <td>56644</td>\n",
       "      <td>ut7dlyh8</td>\n",
       "      <td>A duplex real-time reverse transcriptase polym...</td>\n",
       "      <td>Abstract A duplex TaqMan real-time reverse tra...</td>\n",
       "      <td>2009-10-31</td>\n",
       "      <td>Wang, Heng; Nattanmai, Seela; Kramer, Laura D....</td>\n",
       "      <td>https://doi.org/10.1016/j.diagmicrobio.2009.07...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['protein', 'genetic']</td>\n",
       "      <td>The targets selected for the assay were the se...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>56645 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0  paper_id  \\\n",
       "0               0  zjufx4fo   \n",
       "1               1  ymceytj3   \n",
       "2               2  wzj2glte   \n",
       "3               3  2sfqsfm1   \n",
       "4               4  i0zym7iq   \n",
       "...           ...       ...   \n",
       "56640       56640  rn4gm3sc   \n",
       "56641       56641  l5oecoot   \n",
       "56642       56642  i4acg2ex   \n",
       "56643       56643  rve0w3dg   \n",
       "56644       56644  ut7dlyh8   \n",
       "\n",
       "                                                   title  \\\n",
       "0      Sequence requirements for RNA strand transfer ...   \n",
       "1      Crystal structure of murine sCEACAM1a[1,4]: a ...   \n",
       "2      Synthesis of a novel hepatitis C virus protein...   \n",
       "3      Structure of coronavirus main proteinase revea...   \n",
       "4      Discontinuous and non-discontinuous subgenomic...   \n",
       "...                                                  ...   \n",
       "56640  Structure-Based Stabilization of Non-native Pr...   \n",
       "56641          Immunomodulators as an antimicrobial tool   \n",
       "56642  Discourses of disease: Representations of tube...   \n",
       "56643                           Equine Infectious Anemia   \n",
       "56644  A duplex real-time reverse transcriptase polym...   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "0      Nidovirus subgenomic mRNAs contain a leader se...   2001-12-17   \n",
       "1      CEACAM1 is a member of the carcinoembryonic an...   2002-05-01   \n",
       "2      Hepatitis C virus (HCV) is an important human ...   2001-07-16   \n",
       "3      The key enzyme in coronavirus polyprotein proc...   2002-07-01   \n",
       "4      Arteri-, corona-, toro- and roniviruses are ev...   2002-12-01   \n",
       "...                                                  ...          ...   \n",
       "56640  [Image: see text] Structure-based stabilizatio...   2020-02-27   \n",
       "56641  The spectrum of infectious diseases has shifte...   2006-10-31   \n",
       "56642  Abstract This paper critically examines the wa...   2008-02-29   \n",
       "56643  This article describes clinical signs and tran...   1993-08-31   \n",
       "56644  Abstract A duplex TaqMan real-time reverse tra...   2009-10-31   \n",
       "\n",
       "                                                 authors  \\\n",
       "0      Pasternak, Alexander O.; van den Born, Erwin; ...   \n",
       "1      Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...   \n",
       "2      Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...   \n",
       "3      Anand, Kanchan; Palm, Gottfried J.; Mesters, J...   \n",
       "4      van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...   \n",
       "...                                                  ...   \n",
       "56640  Lin, Shan-Meng; Lin, Shih-Chao; Hsu, Jia-Ning;...   \n",
       "56641           Pirofski, Liise-anne; Casadevall, Arturo   \n",
       "56642  Lawrence, Jody; Kearns, Robin A.; Park, Julie;...   \n",
       "56643                             Sellon, Debra Clabough   \n",
       "56644  Wang, Heng; Nattanmai, Seela; Kramer, Laura D....   \n",
       "\n",
       "                                                     url Statistical Methods  \\\n",
       "0      http://europepmc.org/articles/pmc125340?pdf=re...                  []   \n",
       "1      http://europepmc.org/articles/pmc125375?pdf=re...                  []   \n",
       "2      https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...                  []   \n",
       "3      http://europepmc.org/articles/pmc126080?pdf=re...           ['model']   \n",
       "4      http://europepmc.org/articles/pmc136939?pdf=re...                  []   \n",
       "...                                                  ...                 ...   \n",
       "56640  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                  []   \n",
       "56641          https://doi.org/10.1016/j.mib.2006.08.004           [' tool']   \n",
       "56642    https://doi.org/10.1016/j.socscimed.2007.10.015                  []   \n",
       "56643      https://doi.org/10.1016/s0749-0739(17)30399-1                  []   \n",
       "56644  https://doi.org/10.1016/j.diagmicrobio.2009.07...                  []   \n",
       "\n",
       "                              statistical_method_details  \\\n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3      Molecular modelling and mutagenesis data impli...   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "56640                                                NaN   \n",
       "56641  However, to date only a few immunomodulators h...   \n",
       "56642                                                NaN   \n",
       "56643                                                NaN   \n",
       "56644                                                NaN   \n",
       "\n",
       "                       Study Type  \\\n",
       "0        ['genom', 'mutagenesis']   \n",
       "1                     ['protein']   \n",
       "2            ['protein', 'genom']   \n",
       "3      ['protein', 'mutagenesis']   \n",
       "4                       ['genom']   \n",
       "...                           ...   \n",
       "56640                 ['protein']   \n",
       "56641                          []   \n",
       "56642                          []   \n",
       "56643      ['review', 'clinical']   \n",
       "56644      ['protein', 'genetic']   \n",
       "\n",
       "                                      study_type_details  \\\n",
       "0      Nidovirus subgenomic mRNAs contain a leader se...   \n",
       "1      We discuss the structural basis of virus recep...   \n",
       "2      Its RNA genome codes for a polyprotein, which ...   \n",
       "3      The key enzyme in coronavirus polyprotein proc...   \n",
       "4      During infection, multiple subgenomic (sg) mRN...   \n",
       "...                                                  ...   \n",
       "56640  [Image: see text] Structure-based stabilizatio...   \n",
       "56641                                                NaN   \n",
       "56642                                                NaN   \n",
       "56643  This article describes clinical signs and tran...   \n",
       "56644  The targets selected for the assay were the se...   \n",
       "\n",
       "                                                 results  \\\n",
       "0      Here we present results of a comprehensive co-...   \n",
       "1      Here we report the crystal structure of solubl...   \n",
       "2                                                    NaN   \n",
       "3      The study illustrates that RNA viruses have ev...   \n",
       "4      We show that BEV mRNAs 3–5 lack a leader seque...   \n",
       "...                                                  ...   \n",
       "56640                                                NaN   \n",
       "56641                                                NaN   \n",
       "56642                                                NaN   \n",
       "56643                                                NaN   \n",
       "56644                                                NaN   \n",
       "\n",
       "                                             coronavirus fatality sample_size  \\\n",
       "0                                                    NaN      NaN         NaN   \n",
       "1                                                    NaN      NaN         NaN   \n",
       "2                                                    NaN      NaN         NaN   \n",
       "3                                                    NaN      NaN         NaN   \n",
       "4                                                    NaN      NaN         NaN   \n",
       "...                                                  ...      ...         ...   \n",
       "56640  Here, we identified a non-native interaction i...      NaN         NaN   \n",
       "56641                                                NaN      NaN         NaN   \n",
       "56642                                                NaN      NaN         NaN   \n",
       "56643                                                NaN      NaN         NaN   \n",
       "56644                                                NaN      NaN         NaN   \n",
       "\n",
       "      paper_focus variables_controlled_for  \n",
       "0             NaN                      NaN  \n",
       "1             NaN                      NaN  \n",
       "2             NaN                      NaN  \n",
       "3             NaN                      NaN  \n",
       "4             NaN                      NaN  \n",
       "...           ...                      ...  \n",
       "56640         NaN                      NaN  \n",
       "56641         NaN                      NaN  \n",
       "56642         NaN                      NaN  \n",
       "56643         NaN                      NaN  \n",
       "56644         NaN                      NaN  \n",
       "\n",
       "[56645 rows x 17 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf = pd.read_csv(\"metadata_features.csv\")\n",
    "mf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Unnamed: 0',\n",
       " 'paper_id',\n",
       " 'title',\n",
       " 'abstract',\n",
       " 'publish_time',\n",
       " 'authors',\n",
       " 'url',\n",
       " 'Statistical Methods',\n",
       " 'statistical_method_details',\n",
       " 'Study Type',\n",
       " 'study_type_details',\n",
       " 'results',\n",
       " 'coronavirus',\n",
       " 'fatality',\n",
       " 'sample_size',\n",
       " 'paper_focus',\n",
       " 'variables_controlled_for']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(mf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]                                                                                       26228\n",
       "['protein']                                                                               6699\n",
       "['clinical']                                                                              4373\n",
       "['review']                                                                                2533\n",
       "['genom']                                                                                 1251\n",
       "                                                                                         ...  \n",
       "['review', 'cross section', 'cross sectional', 'clinical']                                   1\n",
       "['retrospective study', 'prospective study', 'clinical']                                     1\n",
       "['genom', 'genetic', 'life cycle', 'clinical']                                               1\n",
       "['prospective cohort', 'prospective cohort', 'prospective observational', 'clinical']        1\n",
       "['protein', 'mutagenesis', 'genetic', 'experimental']                                        1\n",
       "Name: Study Type, Length: 607, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf['Study Type'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Initially, will work with this and see F1 score. Will then create own features by traditional feature engineering and reverse engineering to get cleanest features set possible. Caveat with this preliminary feature set is that when [] is taken out, 46% of the metadata is not accounted for (no study type key words). Can remedy this through changing n-gram feature-set to be more specific and relevant/feature engineering."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Features Preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>Study Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>['genom', 'mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>['protein']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>['protein', 'genom']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>['protein', 'mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>['genom']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56640</th>\n",
       "      <td>rn4gm3sc</td>\n",
       "      <td>['protein']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56641</th>\n",
       "      <td>l5oecoot</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56642</th>\n",
       "      <td>i4acg2ex</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56643</th>\n",
       "      <td>rve0w3dg</td>\n",
       "      <td>['review', 'clinical']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56644</th>\n",
       "      <td>ut7dlyh8</td>\n",
       "      <td>['protein', 'genetic']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>56645 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       paper_id                  Study Type\n",
       "0      zjufx4fo    ['genom', 'mutagenesis']\n",
       "1      ymceytj3                 ['protein']\n",
       "2      wzj2glte        ['protein', 'genom']\n",
       "3      2sfqsfm1  ['protein', 'mutagenesis']\n",
       "4      i0zym7iq                   ['genom']\n",
       "...         ...                         ...\n",
       "56640  rn4gm3sc                 ['protein']\n",
       "56641  l5oecoot                          []\n",
       "56642  i4acg2ex                          []\n",
       "56643  rve0w3dg      ['review', 'clinical']\n",
       "56644  ut7dlyh8      ['protein', 'genetic']\n",
       "\n",
       "[56645 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features = mf.drop(columns = ['Unnamed: 0',\n",
    " 'title',\n",
    " 'abstract',\n",
    " 'publish_time',\n",
    " 'authors',\n",
    " 'url',\n",
    " 'Statistical Methods',\n",
    " 'statistical_method_details',\n",
    " 'study_type_details',\n",
    " 'results',\n",
    " 'coronavirus',\n",
    " 'fatality',\n",
    " 'sample_size',\n",
    " 'paper_focus',\n",
    " 'variables_controlled_for'])\n",
    "features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "paper_id      0\n",
       "Study Type    0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>Study Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>['genom', 'mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>['protein']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>['protein', 'genom']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>['protein', 'mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>['genom']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56638</th>\n",
       "      <td>jna496x7</td>\n",
       "      <td>['protein']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56639</th>\n",
       "      <td>4jg4t8po</td>\n",
       "      <td>['protein']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56640</th>\n",
       "      <td>rn4gm3sc</td>\n",
       "      <td>['protein']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56643</th>\n",
       "      <td>rve0w3dg</td>\n",
       "      <td>['review', 'clinical']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56644</th>\n",
       "      <td>ut7dlyh8</td>\n",
       "      <td>['protein', 'genetic']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30417 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       paper_id                  Study Type\n",
       "0      zjufx4fo    ['genom', 'mutagenesis']\n",
       "1      ymceytj3                 ['protein']\n",
       "2      wzj2glte        ['protein', 'genom']\n",
       "3      2sfqsfm1  ['protein', 'mutagenesis']\n",
       "4      i0zym7iq                   ['genom']\n",
       "...         ...                         ...\n",
       "56638  jna496x7                 ['protein']\n",
       "56639  4jg4t8po                 ['protein']\n",
       "56640  rn4gm3sc                 ['protein']\n",
       "56643  rve0w3dg      ['review', 'clinical']\n",
       "56644  ut7dlyh8      ['protein', 'genetic']\n",
       "\n",
       "[30417 rows x 2 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Get rid of rows with empty lists (STRING TYPE) since void of study type features.\n",
    "# test2 =  features[features['Study Type'].astype(bool)]\n",
    "# test2\n",
    "to_drop = ['[]']\n",
    "features = features[~features['Study Type'].isin(to_drop)]\n",
    "features"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Joining feature set to label set (Kaggle for now)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to attach study type features to labels, need to use CoronaWhy annotations dataset that has cord_uid that matches with paper_id from above as well as sha that matches with id in label dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_cord_uid = pd.read_csv(\"Study_Design_Annotation_Snapshot_4_12_20.csv\")\n",
    "labels = pd.read_csv(\"design.csv\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['paper_id',\n",
       " 'Study Type',\n",
       " 'Assignee',\n",
       " 'cord_uid',\n",
       " 'sha',\n",
       " 'title',\n",
       " 'url',\n",
       " 'in silico',\n",
       " 'in vitro',\n",
       " 'in vivo',\n",
       " 'Systematic reivew or meta-analysis of RCTs',\n",
       " 'RCT',\n",
       " 'Non-randomized controlled trial',\n",
       " 'Comparative study with a concurrent or historical comparator (cohort studies, case-control studies, 2 or more single-arm studies, or interrupted time series with a control group)',\n",
       " 'Descriptive study without a comparator (single-arm pre-post studies, case series, case reports)',\n",
       " 'Systematic review or meta-analysis of studies other than RCTs',\n",
       " 'Other ',\n",
       " 'General Notes']"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(mf_cordid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>lwla5ugt</td>\n",
       "      <td>['genom', 'bioinformatics']</td>\n",
       "      <td>lwla5ugt</td>\n",
       "      <td>8970ef10049816050cb1bc5fbf600865a5996ad4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gdsfkw1b</td>\n",
       "      <td>['protein', 'review']</td>\n",
       "      <td>gdsfkw1b</td>\n",
       "      <td>8d7400a2b387820cd391d7df8194642e50402a0c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>e1iaiwc6</td>\n",
       "      <td>['survey']</td>\n",
       "      <td>e1iaiwc6</td>\n",
       "      <td>eddc547cbba693715e4fd55e6a946e8ec7d96bc2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>o088rxb5</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>o088rxb5</td>\n",
       "      <td>9f63ebfaab049c0968d72166761217b3aaa1fe00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>icp9d998</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>icp9d998</td>\n",
       "      <td>8c640872048570b0a671ec08c83ef6cbebcdd78e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595</th>\n",
       "      <td>f7i4lq2w</td>\n",
       "      <td>['genetic', 'clinical']</td>\n",
       "      <td>f7i4lq2w</td>\n",
       "      <td>450921cad3744aa651260fdd482038f546992999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>hwt1foyu</td>\n",
       "      <td>['protein']</td>\n",
       "      <td>hwt1foyu</td>\n",
       "      <td>ce0f694c2a6f8568c566118e14bef4c1898957f6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>c9zya2t7</td>\n",
       "      <td>['protein']</td>\n",
       "      <td>c9zya2t7</td>\n",
       "      <td>2b7cc2e066ab4f8252246b530424d878b260b108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>598</th>\n",
       "      <td>437zx0xr</td>\n",
       "      <td>['survey']</td>\n",
       "      <td>437zx0xr</td>\n",
       "      <td>9da58647ca34a42efcf47cec929b7163a808a65d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>0y5jaf1m</td>\n",
       "      <td>['protein']</td>\n",
       "      <td>0y5jaf1m</td>\n",
       "      <td>60a342d1b18b2059aa722f46cc9f8b9770b6919c</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>600 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     paper_id                   Study Type  cord_uid  \\\n",
       "0    lwla5ugt  ['genom', 'bioinformatics']  lwla5ugt   \n",
       "1    gdsfkw1b        ['protein', 'review']  gdsfkw1b   \n",
       "2    e1iaiwc6                   ['survey']  e1iaiwc6   \n",
       "3    o088rxb5                 ['clinical']  o088rxb5   \n",
       "4    icp9d998                 ['clinical']  icp9d998   \n",
       "..        ...                          ...       ...   \n",
       "595  f7i4lq2w      ['genetic', 'clinical']  f7i4lq2w   \n",
       "596  hwt1foyu                  ['protein']  hwt1foyu   \n",
       "597  c9zya2t7                  ['protein']  c9zya2t7   \n",
       "598  437zx0xr                   ['survey']  437zx0xr   \n",
       "599  0y5jaf1m                  ['protein']  0y5jaf1m   \n",
       "\n",
       "                                          sha  \n",
       "0    8970ef10049816050cb1bc5fbf600865a5996ad4  \n",
       "1    8d7400a2b387820cd391d7df8194642e50402a0c  \n",
       "2    eddc547cbba693715e4fd55e6a946e8ec7d96bc2  \n",
       "3    9f63ebfaab049c0968d72166761217b3aaa1fe00  \n",
       "4    8c640872048570b0a671ec08c83ef6cbebcdd78e  \n",
       "..                                        ...  \n",
       "595  450921cad3744aa651260fdd482038f546992999  \n",
       "596  ce0f694c2a6f8568c566118e14bef4c1898957f6  \n",
       "597  2b7cc2e066ab4f8252246b530424d878b260b108  \n",
       "598  9da58647ca34a42efcf47cec929b7163a808a65d  \n",
       "599  60a342d1b18b2059aa722f46cc9f8b9770b6919c  \n",
       "\n",
       "[600 rows x 4 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feat_cordid = features.merge(get_cord_uid, how='inner', left_on='paper_id', right_on='cord_uid')\n",
    "feat_cordid = feat_cordid.drop(columns = ['Assignee',\n",
    " 'title',\n",
    " 'url',\n",
    " 'in silico',\n",
    " 'in vitro',\n",
    " 'in vivo',\n",
    " 'Systematic reivew or meta-analysis of RCTs',\n",
    " 'RCT',\n",
    " 'Non-randomized controlled trial',\n",
    " 'Comparative study with a concurrent or historical comparator (cohort studies, case-control studies, 2 or more single-arm studies, or interrupted time series with a control group)',\n",
    " 'Descriptive study without a comparator (single-arm pre-post studies, case series, case reports)',\n",
    " 'Systematic review or meta-analysis of studies other than RCTs',\n",
    " 'Other ',\n",
    " 'General Notes'] )\n",
    "feat_cordid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "      <th>label</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7n75m669</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>7n75m669</td>\n",
       "      <td>5e9ccecdd40825f5c30da3a900fc5cf1b063b47d</td>\n",
       "      <td>8</td>\n",
       "      <td>5e9ccecdd40825f5c30da3a900fc5cf1b063b47d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ie3msqwt</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>ie3msqwt</td>\n",
       "      <td>5b15c62ee761d6eefeb7d2ef023f804bfdc41edc</td>\n",
       "      <td>9</td>\n",
       "      <td>5b15c62ee761d6eefeb7d2ef023f804bfdc41edc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>u1m7yr8f</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>u1m7yr8f</td>\n",
       "      <td>eac85c85c3a2ac1d3fe53e1e3b3fee98f52527ad</td>\n",
       "      <td>7</td>\n",
       "      <td>eac85c85c3a2ac1d3fe53e1e3b3fee98f52527ad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>a27prjq9</td>\n",
       "      <td>['review']</td>\n",
       "      <td>a27prjq9</td>\n",
       "      <td>4bc6ead9368ec84dba7c56b78624f681293e568b</td>\n",
       "      <td>7</td>\n",
       "      <td>4bc6ead9368ec84dba7c56b78624f681293e568b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9vrny92m</td>\n",
       "      <td>['review']</td>\n",
       "      <td>9vrny92m</td>\n",
       "      <td>d7d62e768f4a8eb40ba522eca28e173e5d53cbbc</td>\n",
       "      <td>4</td>\n",
       "      <td>d7d62e768f4a8eb40ba522eca28e173e5d53cbbc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   paper_id                                      Study Type  cord_uid  \\\n",
       "0  7n75m669                                     ['genetic']  7n75m669   \n",
       "1  ie3msqwt                                     ['genetic']  ie3msqwt   \n",
       "2  u1m7yr8f  ['questionnaire', 'survey', 'cross-sectional']  u1m7yr8f   \n",
       "3  a27prjq9                                      ['review']  a27prjq9   \n",
       "4  9vrny92m                                      ['review']  9vrny92m   \n",
       "\n",
       "                                        sha  label  \\\n",
       "0  5e9ccecdd40825f5c30da3a900fc5cf1b063b47d      8   \n",
       "1  5b15c62ee761d6eefeb7d2ef023f804bfdc41edc      9   \n",
       "2  eac85c85c3a2ac1d3fe53e1e3b3fee98f52527ad      7   \n",
       "3  4bc6ead9368ec84dba7c56b78624f681293e568b      7   \n",
       "4  d7d62e768f4a8eb40ba522eca28e173e5d53cbbc      4   \n",
       "\n",
       "                                         id  \n",
       "0  5e9ccecdd40825f5c30da3a900fc5cf1b063b47d  \n",
       "1  5b15c62ee761d6eefeb7d2ef023f804bfdc41edc  \n",
       "2  eac85c85c3a2ac1d3fe53e1e3b3fee98f52527ad  \n",
       "3  4bc6ead9368ec84dba7c56b78624f681293e568b  \n",
       "4  d7d62e768f4a8eb40ba522eca28e173e5d53cbbc  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels_merge = feat_cordid.merge(labels, how='inner', left_on='sha', right_on='id')\n",
    "labels_merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['paper_id', 'Study Type', 'cord_uid', 'sha', 'label', 'id']"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(labels_merge)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7n75m669</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ie3msqwt</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>u1m7yr8f</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>a27prjq9</td>\n",
       "      <td>['review']</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9vrny92m</td>\n",
       "      <td>['review']</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   paper_id                                      Study Type  label\n",
       "0  7n75m669                                     ['genetic']      8\n",
       "1  ie3msqwt                                     ['genetic']      9\n",
       "2  u1m7yr8f  ['questionnaire', 'survey', 'cross-sectional']      7\n",
       "3  a27prjq9                                      ['review']      7\n",
       "4  9vrny92m                                      ['review']      4"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final = labels_merge.drop(columns = ['cord_uid', 'sha', 'id'])\n",
    "final"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Only 5 results. This is a sad. It is due to new dataset being larger (10,000 more papers). So there are going to be more dissimilar paper_ids.  In order to limit dataset reduction, need to see if featuers dataframe can be remade with longer(sha?) paper_id in order to match with labels dataframe directly."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If this does not work out, will then need to synthesize labels. So to summarize:\n",
    "1. Max out matches between feature set and label set\n",
    "2. Feature engineer via revising ngram list/reverse engineer\n",
    "3. Synthetic labels for mutlilabels annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Trying direct merge with mf and kaggle label set via sha."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>url</th>\n",
       "      <th>Statistical Methods</th>\n",
       "      <th>statistical_method_details</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>study_type_details</th>\n",
       "      <th>results</th>\n",
       "      <th>coronavirus</th>\n",
       "      <th>fatality</th>\n",
       "      <th>sample_size</th>\n",
       "      <th>paper_focus</th>\n",
       "      <th>variables_controlled_for</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>b2897e1277f56641193a6db73825f707eed3e4c9</td>\n",
       "      <td>Sequence requirements for RNA strand transfer ...</td>\n",
       "      <td>Nidovirus subgenomic mRNAs contain a leader se...</td>\n",
       "      <td>2001-12-17</td>\n",
       "      <td>Pasternak, Alexander O.; van den Born, Erwin; ...</td>\n",
       "      <td>http://europepmc.org/articles/pmc125340?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "      <td>Here we present results of a comprehensive co-...</td>\n",
       "      <td>Here we present results of a comprehensive co-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>e3d0d482ebd9a8ba81c254cc433f314142e72174</td>\n",
       "      <td>Crystal structure of murine sCEACAM1a[1,4]: a ...</td>\n",
       "      <td>CEACAM1 is a member of the carcinoembryonic an...</td>\n",
       "      <td>2002-05-01</td>\n",
       "      <td>Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...</td>\n",
       "      <td>http://europepmc.org/articles/pmc125375?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Here we report the crystal structure of solubl...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>00b1d99e70f779eb4ede50059db469c65e8c1469</td>\n",
       "      <td>Synthesis of a novel hepatitis C virus protein...</td>\n",
       "      <td>Hepatitis C virus (HCV) is an important human ...</td>\n",
       "      <td>2001-07-16</td>\n",
       "      <td>Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0</td>\n",
       "      <td>Structure of coronavirus main proteinase revea...</td>\n",
       "      <td>The key enzyme in coronavirus polyprotein proc...</td>\n",
       "      <td>2002-07-01</td>\n",
       "      <td>Anand, Kanchan; Palm, Gottfried J.; Mesters, J...</td>\n",
       "      <td>http://europepmc.org/articles/pmc126080?pdf=re...</td>\n",
       "      <td>['model']</td>\n",
       "      <td>Molecular modelling and mutagenesis data impli...</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "      <td>Molecular modelling and mutagenesis data impli...</td>\n",
       "      <td>The study illustrates that RNA viruses have ev...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>dde02f11923815e6a16a31dd6298c46b109c5dfa</td>\n",
       "      <td>Discontinuous and non-discontinuous subgenomic...</td>\n",
       "      <td>Arteri-, corona-, toro- and roniviruses are ev...</td>\n",
       "      <td>2002-12-01</td>\n",
       "      <td>van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...</td>\n",
       "      <td>http://europepmc.org/articles/pmc136939?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>We show that BEV mRNAs 3–5 lack a leader seque...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59125</th>\n",
       "      <td>59125</td>\n",
       "      <td>pwq5gir9</td>\n",
       "      <td>20f185b11fa23da8938efb5c839c6c078c66f1e9</td>\n",
       "      <td>Cutting-Edge Infectious Disease Diagnostics wi...</td>\n",
       "      <td>Three recent Science articles (Chen et al., 20...</td>\n",
       "      <td>2018-06-13</td>\n",
       "      <td>Chiu, Charles</td>\n",
       "      <td>https://doi.org/10.1016/j.chom.2018.05.016</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>Three recent Science articles (Chen et al., 20...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59126</th>\n",
       "      <td>59126</td>\n",
       "      <td>3racv2qe</td>\n",
       "      <td>e84532333db8f219f20ae0a8fe18130c0d010b04</td>\n",
       "      <td>Intranasal administration of oxytocin: Behavio...</td>\n",
       "      <td>Abstract The intranasal (IN-) administration o...</td>\n",
       "      <td>2013-09-30</td>\n",
       "      <td>Veening, Jan G.; Olivier, Berend</td>\n",
       "      <td>https://doi.org/10.1016/j.neubiorev.2013.04.012</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['review']</td>\n",
       "      <td>The present review explores our current knowle...</td>\n",
       "      <td>Only very recently in a microdialysis study in...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59127</th>\n",
       "      <td>59127</td>\n",
       "      <td>jngto9kv</td>\n",
       "      <td>31105078a2953217223699d09c6a80d0f5edfdf6</td>\n",
       "      <td>Infecciones virales graves en pacientes inmuno...</td>\n",
       "      <td>Resumen Los virus tienen un papel importante d...</td>\n",
       "      <td>2011-04-30</td>\n",
       "      <td>Díaz, A.; Zaragoza, R.; Granada, R.; Salavert, M.</td>\n",
       "      <td>https://doi.org/10.1016/j.medin.2011.01.001</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['review']</td>\n",
       "      <td>A review of the literature has been performed ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Furthermore, ventilator-associated pneumonia (...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59128</th>\n",
       "      <td>59128</td>\n",
       "      <td>td0776wj</td>\n",
       "      <td>b1f977d58a0eba25acdee5912ef71c372b0632de</td>\n",
       "      <td>18 Omics, Bioinformatics, and Infectious Disea...</td>\n",
       "      <td>Bioinformatics and the ‘omics’ studies are rap...</td>\n",
       "      <td>2011-12-31</td>\n",
       "      <td>Paszkiewicz, Konrad H.; Giezen, Mark van der</td>\n",
       "      <td>https://doi.org/10.1016/b978-0-12-384890-1.000...</td>\n",
       "      <td>['sequencing', 'methods']</td>\n",
       "      <td>In addition, we provide an overview of methods...</td>\n",
       "      <td>['bioinformatics']</td>\n",
       "      <td>Bioinformatics and the ‘omics’ studies are rap...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59129</th>\n",
       "      <td>59129</td>\n",
       "      <td>jxbr6hbk</td>\n",
       "      <td>4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...</td>\n",
       "      <td>Identifying spatio-temporal dynamics of Ebola ...</td>\n",
       "      <td>Containing the recent West African outbreak of...</td>\n",
       "      <td>2017-11-29</td>\n",
       "      <td>Gustafson, Kyle B.; Proctor, Joshua L.</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>['model', 'forecast', ' tool']</td>\n",
       "      <td>Despite recent progress in analysing and model...</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>Calibrated to phylogenetic linkages of virus g...</td>\n",
       "      <td>Consistent with other investigations, our resu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59130 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0  paper_id  \\\n",
       "0               0  zjufx4fo   \n",
       "1               1  ymceytj3   \n",
       "2               2  wzj2glte   \n",
       "3               3  2sfqsfm1   \n",
       "4               4  i0zym7iq   \n",
       "...           ...       ...   \n",
       "59125       59125  pwq5gir9   \n",
       "59126       59126  3racv2qe   \n",
       "59127       59127  jngto9kv   \n",
       "59128       59128  td0776wj   \n",
       "59129       59129  jxbr6hbk   \n",
       "\n",
       "                                                     sha  \\\n",
       "0               b2897e1277f56641193a6db73825f707eed3e4c9   \n",
       "1               e3d0d482ebd9a8ba81c254cc433f314142e72174   \n",
       "2               00b1d99e70f779eb4ede50059db469c65e8c1469   \n",
       "3               cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   \n",
       "4               dde02f11923815e6a16a31dd6298c46b109c5dfa   \n",
       "...                                                  ...   \n",
       "59125           20f185b11fa23da8938efb5c839c6c078c66f1e9   \n",
       "59126           e84532333db8f219f20ae0a8fe18130c0d010b04   \n",
       "59127           31105078a2953217223699d09c6a80d0f5edfdf6   \n",
       "59128           b1f977d58a0eba25acdee5912ef71c372b0632de   \n",
       "59129  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   \n",
       "\n",
       "                                                   title  \\\n",
       "0      Sequence requirements for RNA strand transfer ...   \n",
       "1      Crystal structure of murine sCEACAM1a[1,4]: a ...   \n",
       "2      Synthesis of a novel hepatitis C virus protein...   \n",
       "3      Structure of coronavirus main proteinase revea...   \n",
       "4      Discontinuous and non-discontinuous subgenomic...   \n",
       "...                                                  ...   \n",
       "59125  Cutting-Edge Infectious Disease Diagnostics wi...   \n",
       "59126  Intranasal administration of oxytocin: Behavio...   \n",
       "59127  Infecciones virales graves en pacientes inmuno...   \n",
       "59128  18 Omics, Bioinformatics, and Infectious Disea...   \n",
       "59129  Identifying spatio-temporal dynamics of Ebola ...   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "0      Nidovirus subgenomic mRNAs contain a leader se...   2001-12-17   \n",
       "1      CEACAM1 is a member of the carcinoembryonic an...   2002-05-01   \n",
       "2      Hepatitis C virus (HCV) is an important human ...   2001-07-16   \n",
       "3      The key enzyme in coronavirus polyprotein proc...   2002-07-01   \n",
       "4      Arteri-, corona-, toro- and roniviruses are ev...   2002-12-01   \n",
       "...                                                  ...          ...   \n",
       "59125  Three recent Science articles (Chen et al., 20...   2018-06-13   \n",
       "59126  Abstract The intranasal (IN-) administration o...   2013-09-30   \n",
       "59127  Resumen Los virus tienen un papel importante d...   2011-04-30   \n",
       "59128  Bioinformatics and the ‘omics’ studies are rap...   2011-12-31   \n",
       "59129  Containing the recent West African outbreak of...   2017-11-29   \n",
       "\n",
       "                                                 authors  \\\n",
       "0      Pasternak, Alexander O.; van den Born, Erwin; ...   \n",
       "1      Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...   \n",
       "2      Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...   \n",
       "3      Anand, Kanchan; Palm, Gottfried J.; Mesters, J...   \n",
       "4      van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...   \n",
       "...                                                  ...   \n",
       "59125                                      Chiu, Charles   \n",
       "59126                   Veening, Jan G.; Olivier, Berend   \n",
       "59127  Díaz, A.; Zaragoza, R.; Granada, R.; Salavert, M.   \n",
       "59128       Paszkiewicz, Konrad H.; Giezen, Mark van der   \n",
       "59129             Gustafson, Kyle B.; Proctor, Joshua L.   \n",
       "\n",
       "                                                     url  \\\n",
       "0      http://europepmc.org/articles/pmc125340?pdf=re...   \n",
       "1      http://europepmc.org/articles/pmc125375?pdf=re...   \n",
       "2      https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...   \n",
       "3      http://europepmc.org/articles/pmc126080?pdf=re...   \n",
       "4      http://europepmc.org/articles/pmc136939?pdf=re...   \n",
       "...                                                  ...   \n",
       "59125         https://doi.org/10.1016/j.chom.2018.05.016   \n",
       "59126    https://doi.org/10.1016/j.neubiorev.2013.04.012   \n",
       "59127        https://doi.org/10.1016/j.medin.2011.01.001   \n",
       "59128  https://doi.org/10.1016/b978-0-12-384890-1.000...   \n",
       "59129  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   \n",
       "\n",
       "                  Statistical Methods  \\\n",
       "0                                  []   \n",
       "1                                  []   \n",
       "2                                  []   \n",
       "3                           ['model']   \n",
       "4                                  []   \n",
       "...                               ...   \n",
       "59125                              []   \n",
       "59126                              []   \n",
       "59127                              []   \n",
       "59128       ['sequencing', 'methods']   \n",
       "59129  ['model', 'forecast', ' tool']   \n",
       "\n",
       "                              statistical_method_details          Study Type  \\\n",
       "0                                                    NaN     ['mutagenesis']   \n",
       "1                                                    NaN                  []   \n",
       "2                                                    NaN                  []   \n",
       "3      Molecular modelling and mutagenesis data impli...     ['mutagenesis']   \n",
       "4                                                    NaN                  []   \n",
       "...                                                  ...                 ...   \n",
       "59125                                                NaN        ['clinical']   \n",
       "59126                                                NaN          ['review']   \n",
       "59127                                                NaN          ['review']   \n",
       "59128  In addition, we provide an overview of methods...  ['bioinformatics']   \n",
       "59129  Despite recent progress in analysing and model...         ['genetic']   \n",
       "\n",
       "                                      study_type_details  \\\n",
       "0      Here we present results of a comprehensive co-...   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3      Molecular modelling and mutagenesis data impli...   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "59125  Three recent Science articles (Chen et al., 20...   \n",
       "59126  The present review explores our current knowle...   \n",
       "59127  A review of the literature has been performed ...   \n",
       "59128  Bioinformatics and the ‘omics’ studies are rap...   \n",
       "59129  Calibrated to phylogenetic linkages of virus g...   \n",
       "\n",
       "                                                 results coronavirus  \\\n",
       "0      Here we present results of a comprehensive co-...         NaN   \n",
       "1      Here we report the crystal structure of solubl...         NaN   \n",
       "2                                                    NaN         NaN   \n",
       "3      The study illustrates that RNA viruses have ev...         NaN   \n",
       "4      We show that BEV mRNAs 3–5 lack a leader seque...         NaN   \n",
       "...                                                  ...         ...   \n",
       "59125                                                NaN         NaN   \n",
       "59126  Only very recently in a microdialysis study in...         NaN   \n",
       "59127                                                NaN         NaN   \n",
       "59128                                                NaN         NaN   \n",
       "59129  Consistent with other investigations, our resu...         NaN   \n",
       "\n",
       "                                                fatality sample_size  \\\n",
       "0                                                    NaN         NaN   \n",
       "1                                                    NaN         NaN   \n",
       "2                                                    NaN         NaN   \n",
       "3                                                    NaN         NaN   \n",
       "4                                                    NaN         NaN   \n",
       "...                                                  ...         ...   \n",
       "59125                                                NaN         NaN   \n",
       "59126                                                NaN         NaN   \n",
       "59127  Furthermore, ventilator-associated pneumonia (...         NaN   \n",
       "59128                                                NaN         NaN   \n",
       "59129                                                NaN         NaN   \n",
       "\n",
       "      paper_focus variables_controlled_for  \n",
       "0             NaN                      NaN  \n",
       "1             NaN                      NaN  \n",
       "2             NaN                      NaN  \n",
       "3             NaN                      NaN  \n",
       "4             NaN                      NaN  \n",
       "...           ...                      ...  \n",
       "59125         NaN                      NaN  \n",
       "59126         NaN                      NaN  \n",
       "59127         NaN                      NaN  \n",
       "59128         NaN                      NaN  \n",
       "59129         NaN                      NaN  \n",
       "\n",
       "[59130 rows x 18 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf2 = pd.read_csv('metadata_features-2.csv')\n",
    "mf2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Unnamed: 0',\n",
       " 'paper_id',\n",
       " 'sha',\n",
       " 'title',\n",
       " 'abstract',\n",
       " 'publish_time',\n",
       " 'authors',\n",
       " 'url',\n",
       " 'Statistical Methods',\n",
       " 'statistical_method_details',\n",
       " 'Study Type',\n",
       " 'study_type_details',\n",
       " 'results',\n",
       " 'coronavirus',\n",
       " 'fatality',\n",
       " 'sample_size',\n",
       " 'paper_focus',\n",
       " 'variables_controlled_for']"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(mf2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>b2897e1277f56641193a6db73825f707eed3e4c9</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>e3d0d482ebd9a8ba81c254cc433f314142e72174</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>00b1d99e70f779eb4ede50059db469c65e8c1469</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>dde02f11923815e6a16a31dd6298c46b109c5dfa</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59125</th>\n",
       "      <td>pwq5gir9</td>\n",
       "      <td>20f185b11fa23da8938efb5c839c6c078c66f1e9</td>\n",
       "      <td>['clinical']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59126</th>\n",
       "      <td>3racv2qe</td>\n",
       "      <td>e84532333db8f219f20ae0a8fe18130c0d010b04</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59127</th>\n",
       "      <td>jngto9kv</td>\n",
       "      <td>31105078a2953217223699d09c6a80d0f5edfdf6</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59128</th>\n",
       "      <td>td0776wj</td>\n",
       "      <td>b1f977d58a0eba25acdee5912ef71c372b0632de</td>\n",
       "      <td>['bioinformatics']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59129</th>\n",
       "      <td>jxbr6hbk</td>\n",
       "      <td>4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59130 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       paper_id                                                sha  \\\n",
       "0      zjufx4fo           b2897e1277f56641193a6db73825f707eed3e4c9   \n",
       "1      ymceytj3           e3d0d482ebd9a8ba81c254cc433f314142e72174   \n",
       "2      wzj2glte           00b1d99e70f779eb4ede50059db469c65e8c1469   \n",
       "3      2sfqsfm1           cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   \n",
       "4      i0zym7iq           dde02f11923815e6a16a31dd6298c46b109c5dfa   \n",
       "...         ...                                                ...   \n",
       "59125  pwq5gir9           20f185b11fa23da8938efb5c839c6c078c66f1e9   \n",
       "59126  3racv2qe           e84532333db8f219f20ae0a8fe18130c0d010b04   \n",
       "59127  jngto9kv           31105078a2953217223699d09c6a80d0f5edfdf6   \n",
       "59128  td0776wj           b1f977d58a0eba25acdee5912ef71c372b0632de   \n",
       "59129  jxbr6hbk  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   \n",
       "\n",
       "               Study Type  \n",
       "0         ['mutagenesis']  \n",
       "1                      []  \n",
       "2                      []  \n",
       "3         ['mutagenesis']  \n",
       "4                      []  \n",
       "...                   ...  \n",
       "59125        ['clinical']  \n",
       "59126          ['review']  \n",
       "59127          ['review']  \n",
       "59128  ['bioinformatics']  \n",
       "59129         ['genetic']  \n",
       "\n",
       "[59130 rows x 3 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features2 = mf2.drop(columns = ['Unnamed: 0',\n",
    " 'title',\n",
    " 'abstract',\n",
    " 'publish_time',\n",
    " 'authors',\n",
    " 'url',\n",
    " 'Statistical Methods',\n",
    " 'statistical_method_details',\n",
    " 'study_type_details',\n",
    " 'results',\n",
    " 'coronavirus',\n",
    " 'fatality',\n",
    " 'sample_size',\n",
    " 'paper_focus',\n",
    " 'variables_controlled_for'])\n",
    "features2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>b2897e1277f56641193a6db73825f707eed3e4c9</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>gi6uaa83</td>\n",
       "      <td>8ae137c8da1607b3a8e4c946c07ca8bda67f88ac</td>\n",
       "      <td>['genetic', 'histories']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0m32ecnu</td>\n",
       "      <td>23bc55d6f63fab18b02004483888db2b6a0bfa48</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>fy4w7xz8</td>\n",
       "      <td>0104f6ceccf92ae8567a0102f89cbb976969a774</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59125</th>\n",
       "      <td>pwq5gir9</td>\n",
       "      <td>20f185b11fa23da8938efb5c839c6c078c66f1e9</td>\n",
       "      <td>['clinical']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59126</th>\n",
       "      <td>3racv2qe</td>\n",
       "      <td>e84532333db8f219f20ae0a8fe18130c0d010b04</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59127</th>\n",
       "      <td>jngto9kv</td>\n",
       "      <td>31105078a2953217223699d09c6a80d0f5edfdf6</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59128</th>\n",
       "      <td>td0776wj</td>\n",
       "      <td>b1f977d58a0eba25acdee5912ef71c372b0632de</td>\n",
       "      <td>['bioinformatics']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59129</th>\n",
       "      <td>jxbr6hbk</td>\n",
       "      <td>4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21864 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       paper_id                                                sha  \\\n",
       "0      zjufx4fo           b2897e1277f56641193a6db73825f707eed3e4c9   \n",
       "3      2sfqsfm1           cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   \n",
       "6      gi6uaa83           8ae137c8da1607b3a8e4c946c07ca8bda67f88ac   \n",
       "7      0m32ecnu           23bc55d6f63fab18b02004483888db2b6a0bfa48   \n",
       "10     fy4w7xz8           0104f6ceccf92ae8567a0102f89cbb976969a774   \n",
       "...         ...                                                ...   \n",
       "59125  pwq5gir9           20f185b11fa23da8938efb5c839c6c078c66f1e9   \n",
       "59126  3racv2qe           e84532333db8f219f20ae0a8fe18130c0d010b04   \n",
       "59127  jngto9kv           31105078a2953217223699d09c6a80d0f5edfdf6   \n",
       "59128  td0776wj           b1f977d58a0eba25acdee5912ef71c372b0632de   \n",
       "59129  jxbr6hbk  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   \n",
       "\n",
       "                     Study Type  \n",
       "0               ['mutagenesis']  \n",
       "3               ['mutagenesis']  \n",
       "6      ['genetic', 'histories']  \n",
       "7                   ['genetic']  \n",
       "10                  ['genetic']  \n",
       "...                         ...  \n",
       "59125              ['clinical']  \n",
       "59126                ['review']  \n",
       "59127                ['review']  \n",
       "59128        ['bioinformatics']  \n",
       "59129               ['genetic']  \n",
       "\n",
       "[21864 rows x 3 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "to_drop = ['[]']\n",
    "features2 = features2[~features2['Study Type'].isin(to_drop)]\n",
    "features2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ub9dtgup</td>\n",
       "      <td>78326020e9f7cd75415a3a192729ccf0c0930281</td>\n",
       "      <td>['genetic', 'questionnaire']</td>\n",
       "      <td>8</td>\n",
       "      <td>78326020e9f7cd75415a3a192729ccf0c0930281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8f0pn321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['survey']</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>xavtxl4m</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['survey']</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ciuhyhb8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3660</th>\n",
       "      <td>l4649du3</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>2</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3661</th>\n",
       "      <td>3vpoxfnv</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>10</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3662</th>\n",
       "      <td>p8vhi6d9</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>['experimental']</td>\n",
       "      <td>2</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3663</th>\n",
       "      <td>a6i73v0o</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3664</th>\n",
       "      <td>nbbcacy4</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>['case study']</td>\n",
       "      <td>9</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3665 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      paper_id                                       sha  \\\n",
       "0     47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1     ub9dtgup  78326020e9f7cd75415a3a192729ccf0c0930281   \n",
       "2     8f0pn321                                       NaN   \n",
       "3     xavtxl4m                                       NaN   \n",
       "4     ciuhyhb8                                       NaN   \n",
       "...        ...                                       ...   \n",
       "3660  l4649du3  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e   \n",
       "3661  3vpoxfnv  eb720970f362e4416b73393f230c8606c837c6dc   \n",
       "3662  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "3663  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "3664  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                          Study Type  label  \\\n",
       "0                                         ['review']      1   \n",
       "1                       ['genetic', 'questionnaire']      8   \n",
       "2                                         ['survey']      2   \n",
       "3                                         ['survey']      2   \n",
       "4                                       ['clinical']      2   \n",
       "...                                              ...    ...   \n",
       "3660                                    ['clinical']      2   \n",
       "3661                                    ['clinical']     10   \n",
       "3662                                ['experimental']      2   \n",
       "3663  ['questionnaire', 'survey', 'cross-sectional']      7   \n",
       "3664                                  ['case study']      9   \n",
       "\n",
       "                                            id  \n",
       "0     6a3fa8ed278df0d05c5e009521de11c72308f60b  \n",
       "1     78326020e9f7cd75415a3a192729ccf0c0930281  \n",
       "2                                          NaN  \n",
       "3                                          NaN  \n",
       "4                                          NaN  \n",
       "...                                        ...  \n",
       "3660  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e  \n",
       "3661  eb720970f362e4416b73393f230c8606c837c6dc  \n",
       "3662  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb  \n",
       "3663  95d9bac431797e2cf8b7294a648ceec3bd50c84b  \n",
       "3664  76edb6d0bd7c47765d3a89032aa610f410506988  \n",
       "\n",
       "[3665 rows x 5 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels_merge2 = features2.merge(labels, how='inner', left_on='sha', right_on='id')\n",
    "labels_merge2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check for duplicates. (There are probably duplicates because sha is not unique identifier but indicator of change)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "429"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels_merge2.sha.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3665"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels_merge2.paper_id.nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are 429 shared papers between sha and id. However, what produces the 3665 results for labels_merge2 is the paper_id column that there are  sha dupilicates. So if we remove duplicates, will get 429 results for modeling."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label_x</th>\n",
       "      <th>id_x</th>\n",
       "      <th>label_y</th>\n",
       "      <th>id_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>1</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ub9dtgup</td>\n",
       "      <td>78326020e9f7cd75415a3a192729ccf0c0930281</td>\n",
       "      <td>['genetic', 'questionnaire']</td>\n",
       "      <td>8</td>\n",
       "      <td>78326020e9f7cd75415a3a192729ccf0c0930281</td>\n",
       "      <td>8</td>\n",
       "      <td>78326020e9f7cd75415a3a192729ccf0c0930281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8f0pn321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['survey']</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>9</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>9</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7n75m669</td>\n",
       "      <td>5e9ccecdd40825f5c30da3a900fc5cf1b063b47d</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>8</td>\n",
       "      <td>5e9ccecdd40825f5c30da3a900fc5cf1b063b47d</td>\n",
       "      <td>8</td>\n",
       "      <td>5e9ccecdd40825f5c30da3a900fc5cf1b063b47d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>l4649du3</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>2</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "      <td>2</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>3vpoxfnv</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>10</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "      <td>10</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>p8vhi6d9</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>['experimental']</td>\n",
       "      <td>2</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>2</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>a6i73v0o</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>7</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>nbbcacy4</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>['case study']</td>\n",
       "      <td>9</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>9</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>430 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     paper_id                                       sha  \\\n",
       "0    47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1    ub9dtgup  78326020e9f7cd75415a3a192729ccf0c0930281   \n",
       "2    8f0pn321                                       NaN   \n",
       "3    9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "4    7n75m669  5e9ccecdd40825f5c30da3a900fc5cf1b063b47d   \n",
       "..        ...                                       ...   \n",
       "425  l4649du3  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e   \n",
       "426  3vpoxfnv  eb720970f362e4416b73393f230c8606c837c6dc   \n",
       "427  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "428  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "429  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                         Study Type  label_x  \\\n",
       "0                                        ['review']        1   \n",
       "1                      ['genetic', 'questionnaire']        8   \n",
       "2                                        ['survey']        2   \n",
       "3              ['questionnaire', 'cross-sectional']        9   \n",
       "4                                       ['genetic']        8   \n",
       "..                                              ...      ...   \n",
       "425                                    ['clinical']        2   \n",
       "426                                    ['clinical']       10   \n",
       "427                                ['experimental']        2   \n",
       "428  ['questionnaire', 'survey', 'cross-sectional']        7   \n",
       "429                                  ['case study']        9   \n",
       "\n",
       "                                         id_x  label_y  \\\n",
       "0    6a3fa8ed278df0d05c5e009521de11c72308f60b        1   \n",
       "1    78326020e9f7cd75415a3a192729ccf0c0930281        8   \n",
       "2                                         NaN        2   \n",
       "3    eb4b8f43f68db5e493984b7e4e56d9f804d7e885        9   \n",
       "4    5e9ccecdd40825f5c30da3a900fc5cf1b063b47d        8   \n",
       "..                                        ...      ...   \n",
       "425  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e        2   \n",
       "426  eb720970f362e4416b73393f230c8606c837c6dc       10   \n",
       "427  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb        2   \n",
       "428  95d9bac431797e2cf8b7294a648ceec3bd50c84b        7   \n",
       "429  76edb6d0bd7c47765d3a89032aa610f410506988        9   \n",
       "\n",
       "                                         id_y  \n",
       "0    6a3fa8ed278df0d05c5e009521de11c72308f60b  \n",
       "1    78326020e9f7cd75415a3a192729ccf0c0930281  \n",
       "2                                         NaN  \n",
       "3    eb4b8f43f68db5e493984b7e4e56d9f804d7e885  \n",
       "4    5e9ccecdd40825f5c30da3a900fc5cf1b063b47d  \n",
       "..                                        ...  \n",
       "425  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e  \n",
       "426  eb720970f362e4416b73393f230c8606c837c6dc  \n",
       "427  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb  \n",
       "428  95d9bac431797e2cf8b7294a648ceec3bd50c84b  \n",
       "429  76edb6d0bd7c47765d3a89032aa610f410506988  \n",
       "\n",
       "[430 rows x 7 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels_merge2 = labels_merge2.drop_duplicates(subset=['sha'])\n",
    "\n",
    "final = pd.merge(labels_merge2, labels, how='inner', left_on='sha', right_on='id')\n",
    "final"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is a really small dataset. So next step is to augment dataset with CoronaWhy dataset as well as Pubmed annotations."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So need to augment annotations dataset (To Kaggle's add, CoronaWhy's) by using Christine Chen's mapping."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1188 entries, 0 to 1187\n",
      "Data columns (total 2 columns):\n",
      " #   Column  Non-Null Count  Dtype \n",
      "---  ------  --------------  ----- \n",
      " 0   label   1188 non-null   int64 \n",
      " 1   id      1187 non-null   object\n",
      "dtypes: int64(1), object(1)\n",
      "memory usage: 18.7+ KB\n"
     ]
    }
   ],
   "source": [
    "labels.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1438 entries, 0 to 1437\n",
      "Data columns (total 2 columns):\n",
      " #   Column  Non-Null Count  Dtype \n",
      "---  ------  --------------  ----- \n",
      " 0   label   1438 non-null   int64 \n",
      " 1   id      1438 non-null   object\n",
      "dtypes: int64(1), object(1)\n",
      "memory usage: 22.6+ KB\n"
     ]
    }
   ],
   "source": [
    "labels2 = pd.read_csv('design-2.csv')\n",
    "labels2.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>8</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnu8gw6w</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>['time series', 'clinical']</td>\n",
       "      <td>5</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ulk3euw</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>['cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>loowpch7</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>['prospective cohort', 'prospective cohort']</td>\n",
       "      <td>4</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>dycx8i7h</td>\n",
       "      <td>2d0d272a793b99ef896547024c51f486e9552e5a</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>0</td>\n",
       "      <td>2d0d272a793b99ef896547024c51f486e9552e5a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>p8vhi6d9</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>['experimental']</td>\n",
       "      <td>2</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>a6i73v0o</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>i2vg8zqc</td>\n",
       "      <td>e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802</td>\n",
       "      <td>['experimental', 'clinical']</td>\n",
       "      <td>0</td>\n",
       "      <td>e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>nbbcacy4</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>['case study']</td>\n",
       "      <td>8</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>498 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     paper_id                                       sha  \\\n",
       "0    47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1    9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2    hnu8gw6w  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3    5ulk3euw  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4    loowpch7  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "..        ...                                       ...   \n",
       "493  dycx8i7h  2d0d272a793b99ef896547024c51f486e9552e5a   \n",
       "494  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "495  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "496  i2vg8zqc  e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802   \n",
       "497  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                         Study Type  label  \\\n",
       "0                                        ['review']      1   \n",
       "1              ['questionnaire', 'cross-sectional']      8   \n",
       "2                       ['time series', 'clinical']      5   \n",
       "3                               ['cross-sectional']      7   \n",
       "4      ['prospective cohort', 'prospective cohort']      4   \n",
       "..                                              ...    ...   \n",
       "493                                     ['genetic']      0   \n",
       "494                                ['experimental']      2   \n",
       "495  ['questionnaire', 'survey', 'cross-sectional']      7   \n",
       "496                    ['experimental', 'clinical']      0   \n",
       "497                                  ['case study']      8   \n",
       "\n",
       "                                           id  \n",
       "0    6a3fa8ed278df0d05c5e009521de11c72308f60b  \n",
       "1    eb4b8f43f68db5e493984b7e4e56d9f804d7e885  \n",
       "2    2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9  \n",
       "3    9f7ed2037968c63fa97eb6fc7c35e9738a654541  \n",
       "4    de707f97b534ca340482600ed1f008b20d7b1b8e  \n",
       "..                                        ...  \n",
       "493  2d0d272a793b99ef896547024c51f486e9552e5a  \n",
       "494  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb  \n",
       "495  95d9bac431797e2cf8b7294a648ceec3bd50c84b  \n",
       "496  e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802  \n",
       "497  76edb6d0bd7c47765d3a89032aa610f410506988  \n",
       "\n",
       "[498 rows x 5 columns]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final2 = pd.merge(features2, labels2, how='inner', left_on='sha', right_on='id')\n",
    "final2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnu8gw6w</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>['time series', 'clinical']</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ulk3euw</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>['cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>loowpch7</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>['prospective cohort', 'prospective cohort']</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>dycx8i7h</td>\n",
       "      <td>2d0d272a793b99ef896547024c51f486e9552e5a</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>p8vhi6d9</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>['experimental']</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>a6i73v0o</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>i2vg8zqc</td>\n",
       "      <td>e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802</td>\n",
       "      <td>['experimental', 'clinical']</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>nbbcacy4</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>['case study']</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>498 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     paper_id                                       sha  \\\n",
       "0    47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1    9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2    hnu8gw6w  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3    5ulk3euw  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4    loowpch7  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "..        ...                                       ...   \n",
       "493  dycx8i7h  2d0d272a793b99ef896547024c51f486e9552e5a   \n",
       "494  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "495  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "496  i2vg8zqc  e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802   \n",
       "497  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                         Study Type  label  \n",
       "0                                        ['review']      1  \n",
       "1              ['questionnaire', 'cross-sectional']      8  \n",
       "2                       ['time series', 'clinical']      5  \n",
       "3                               ['cross-sectional']      7  \n",
       "4      ['prospective cohort', 'prospective cohort']      4  \n",
       "..                                              ...    ...  \n",
       "493                                     ['genetic']      0  \n",
       "494                                ['experimental']      2  \n",
       "495  ['questionnaire', 'survey', 'cross-sectional']      7  \n",
       "496                    ['experimental', 'clinical']      0  \n",
       "497                                  ['case study']      8  \n",
       "\n",
       "[498 rows x 4 columns]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final2 = final2.drop(columns = ['id'])\n",
    "final2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "69 more Kaggle labels with updated Kaggle dataset merged with Kriti's dataset."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Also, need to use updated CoronaWhy dataset. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "cw_labels = pd.read_csv('CoY_newlist.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Assignee',\n",
       " 'cord_uid',\n",
       " 'sha',\n",
       " 'title',\n",
       " 'url',\n",
       " 'Computational ',\n",
       " 'Experimental-in vitro',\n",
       " 'Experimental-in vivo',\n",
       " 'Systematic review and/or meta-analysis',\n",
       " 'Clinical-interventional',\n",
       " 'Clinical-interventional.1',\n",
       " 'Clinical-observational',\n",
       " 'Clinical-observational.1',\n",
       " 'Systematic review and/or meta-analysis ',\n",
       " 'Other',\n",
       " 'General Notes']"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(cw_labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "cw_labels = cw_labels.drop(columns = ['Other', 'General Notes'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Rename/Clean-up labels\n",
    "cw_labels = cw_labels.rename(columns={'Computational ': 'Computational', 'Experimental-in vitro' : 'Experimental - in vitro', 'Experimental-in vivo' : 'Experimental - in vivo', 'Clinical-interventional.1' : 'Clinical-interventional', 'Clinical-observational.1' : 'Clinical-observational', 'Systematic review and/or meta-analysis ' : 'Systematic review and/or meta-analysis'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Assignee',\n",
       " 'cord_uid',\n",
       " 'sha',\n",
       " 'title',\n",
       " 'url',\n",
       " 'Computational',\n",
       " 'Experimental - in vitro',\n",
       " 'Experimental - in vivo',\n",
       " 'Systematic review and/or meta-analysis',\n",
       " 'Clinical-interventional',\n",
       " 'Clinical-interventional',\n",
       " 'Clinical-observational',\n",
       " 'Clinical-observational',\n",
       " 'Systematic review and/or meta-analysis']"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(cw_labels)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mapping Kaggle to CoY_newlist annotations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 8, 5, 7, 4, 2, 9, 6, 0, 3])"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final2.explode('label')\n",
    "final2['label'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic_map={\n",
    "         1:'Systematic review and/or meta-analysis',\n",
    "         2:'Clinical-interventional',\n",
    "         3:'Clinical-interventional',\n",
    "         4:'Clinical-observational',\n",
    "         5:'Clinical-observational',\n",
    "         6:'Clinical-observational',\n",
    "         7:'Clinical-observational',\n",
    "         8:'Clinical-observational',\n",
    "         9:'Computational',\n",
    "         0:'Other'\n",
    "        }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "      <th>mapped</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>Systematic review and/or meta-analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>8</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnu8gw6w</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>['time series', 'clinical']</td>\n",
       "      <td>5</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ulk3euw</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>['cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>loowpch7</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>['prospective cohort', 'prospective cohort']</td>\n",
       "      <td>4</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   paper_id                                       sha  \\\n",
       "0  47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1  9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2  hnu8gw6w  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3  5ulk3euw  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4  loowpch7  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "\n",
       "                                     Study Type  label  \\\n",
       "0                                    ['review']      1   \n",
       "1          ['questionnaire', 'cross-sectional']      8   \n",
       "2                   ['time series', 'clinical']      5   \n",
       "3                           ['cross-sectional']      7   \n",
       "4  ['prospective cohort', 'prospective cohort']      4   \n",
       "\n",
       "                                   mapped  \n",
       "0  Systematic review and/or meta-analysis  \n",
       "1                  Clinical-observational  \n",
       "2                  Clinical-observational  \n",
       "3                  Clinical-observational  \n",
       "4                  Clinical-observational  "
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final2['mapped'] = final2.apply(lambda x: dic_map[x.label], axis = 1) \n",
    "final2.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Remove all rows that have 'Other' in Mapped column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 498 entries, 0 to 497\n",
      "Data columns (total 5 columns):\n",
      " #   Column      Non-Null Count  Dtype \n",
      "---  ------      --------------  ----- \n",
      " 0   paper_id    498 non-null    object\n",
      " 1   sha         498 non-null    object\n",
      " 2   Study Type  498 non-null    object\n",
      " 3   label       498 non-null    int64 \n",
      " 4   mapped      498 non-null    object\n",
      "dtypes: int64(1), object(4)\n",
      "memory usage: 23.3+ KB\n"
     ]
    }
   ],
   "source": [
    "final2.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "      <th>mapped</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>Systematic review and/or meta-analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>8</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnu8gw6w</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>['time series', 'clinical']</td>\n",
       "      <td>5</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ulk3euw</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>['cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>loowpch7</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>['prospective cohort', 'prospective cohort']</td>\n",
       "      <td>4</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>l4649du3</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>2</td>\n",
       "      <td>Clinical-interventional</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>3vpoxfnv</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>9</td>\n",
       "      <td>Computational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>p8vhi6d9</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>['experimental']</td>\n",
       "      <td>2</td>\n",
       "      <td>Clinical-interventional</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>a6i73v0o</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>nbbcacy4</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>['case study']</td>\n",
       "      <td>8</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>398 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     paper_id                                       sha  \\\n",
       "0    47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1    9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2    hnu8gw6w  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3    5ulk3euw  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4    loowpch7  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "..        ...                                       ...   \n",
       "490  l4649du3  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e   \n",
       "491  3vpoxfnv  eb720970f362e4416b73393f230c8606c837c6dc   \n",
       "494  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "495  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "497  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                         Study Type  label  \\\n",
       "0                                        ['review']      1   \n",
       "1              ['questionnaire', 'cross-sectional']      8   \n",
       "2                       ['time series', 'clinical']      5   \n",
       "3                               ['cross-sectional']      7   \n",
       "4      ['prospective cohort', 'prospective cohort']      4   \n",
       "..                                              ...    ...   \n",
       "490                                    ['clinical']      2   \n",
       "491                                    ['clinical']      9   \n",
       "494                                ['experimental']      2   \n",
       "495  ['questionnaire', 'survey', 'cross-sectional']      7   \n",
       "497                                  ['case study']      8   \n",
       "\n",
       "                                     mapped  \n",
       "0    Systematic review and/or meta-analysis  \n",
       "1                    Clinical-observational  \n",
       "2                    Clinical-observational  \n",
       "3                    Clinical-observational  \n",
       "4                    Clinical-observational  \n",
       "..                                      ...  \n",
       "490                 Clinical-interventional  \n",
       "491                           Computational  \n",
       "494                 Clinical-interventional  \n",
       "495                  Clinical-observational  \n",
       "497                  Clinical-observational  \n",
       "\n",
       "[398 rows x 5 columns]"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final2 = final2[final2.label != 0]\n",
    "final2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "100 less annotations is a huge drop. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>is_rct</th>\n",
       "      <th>is_systematic_reivew</th>\n",
       "      <th>is_review</th>\n",
       "      <th>is_meta_analysis</th>\n",
       "      <th>is_observational_study</th>\n",
       "      <th>is_case_series</th>\n",
       "      <th>is_clincal_trial</th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>pubmed_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>11742998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>11980704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>11447125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>12093723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>12456663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57361</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>rn4gm3sc</td>\n",
       "      <td>32105468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57362</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>l5oecoot</td>\n",
       "      <td>16931122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57363</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>i4acg2ex</td>\n",
       "      <td>18035462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57364</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>rve0w3dg</td>\n",
       "      <td>8395326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57365</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>ut7dlyh8</td>\n",
       "      <td>19748425</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>57366 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       is_rct  is_systematic_reivew  is_review  is_meta_analysis  \\\n",
       "0       False                 False      False             False   \n",
       "1       False                 False      False             False   \n",
       "2       False                 False      False             False   \n",
       "3       False                 False      False             False   \n",
       "4       False                 False      False             False   \n",
       "...       ...                   ...        ...               ...   \n",
       "57361   False                 False      False             False   \n",
       "57362   False                 False       True             False   \n",
       "57363   False                 False      False             False   \n",
       "57364   False                 False       True             False   \n",
       "57365   False                 False      False             False   \n",
       "\n",
       "       is_observational_study  is_case_series  is_clincal_trial  cord_uid  \\\n",
       "0                       False           False             False  zjufx4fo   \n",
       "1                       False           False             False  ymceytj3   \n",
       "2                       False           False             False  wzj2glte   \n",
       "3                       False           False             False  2sfqsfm1   \n",
       "4                       False           False             False  i0zym7iq   \n",
       "...                       ...             ...               ...       ...   \n",
       "57361                   False           False             False  rn4gm3sc   \n",
       "57362                   False           False             False  l5oecoot   \n",
       "57363                   False           False             False  i4acg2ex   \n",
       "57364                   False           False             False  rve0w3dg   \n",
       "57365                   False           False             False  ut7dlyh8   \n",
       "\n",
       "       pubmed_id  \n",
       "0       11742998  \n",
       "1       11980704  \n",
       "2       11447125  \n",
       "3       12093723  \n",
       "4       12456663  \n",
       "...          ...  \n",
       "57361   32105468  \n",
       "57362   16931122  \n",
       "57363   18035462  \n",
       "57364    8395326  \n",
       "57365   19748425  \n",
       "\n",
       "[57366 rows x 9 columns]"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed_labels = pd.read_csv('metadata_extra_info_from_pubmed_db.csv')\n",
    "pubmed_labels"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MAPPING SCHEME\n",
    "\n",
    "Mapping Kaggle to new list\n",
    "1 - Systematic review -> Systematic review and/or meta-analysis\n",
    "2 - Randomized control trial -> Clinical-interventional\n",
    "3 - Non-randomized trial -> Clinical - interventional\n",
    "4 - Prospective observational -> Clinical -observational\n",
    "5 - Time-to-event analysis -> Clinical-observational\n",
    "6 - Retrospective observational -> Clinical-observational\n",
    "7 - Cross-sectional -> Clinical-observational\n",
    "8 - Case series -> Clinical-observational\n",
    "9 - Modeling -> Computational\n",
    "0 - Other -> drop\n",
    "5:48\n",
    "Mapping our data to new list\n",
    "In silico -> Computational\n",
    "In vitro  -> Experimental - in vitro\n",
    "In vivo -> Experimental - in vivo\n",
    "Systematic review or meta-analysis of RCTs -> Systematic review and/or meta-analysis\n",
    "Randomized controlled trial -> Clinical-interventional\n",
    "Non-randomized controlled trial -> Clinical-interventional\n",
    "Comparative study  -> Clinical -observational\t\n",
    "Descriptive study  -> Clinical -observational\n",
    "Systematic review or meta-analysis of studies other than RCTs -> Systematic review and/or meta-analysis\n",
    "Other (commentary, editorial, non-systematic review, expert opinion) -> Other\n",
    "(edited)\n",
    "5:53\n",
    "Also @Dan Sosa and I are doing quality check on our labels so there might be an updated version coming up in the next few days. A major update will be an addition of a new category “Review”, which deals with non-systematic literature reviews\n",
    "5:55\n",
    "FYI, here’s the new list of categories\n",
    "Computational studies (simulation, modeling)\n",
    "Experimental studies (animals, cells, perhaps similar to molecular studies in  Artur’s list)\n",
    "2.1 In vitro\n",
    "2.2 In vivo\n",
    "3. Clinical studies (human subjects)\n",
    "3.1 Interventional studies (RCT, CCT, Phase 1-4 clinical trials)\n",
    "3.2 Observational studies (prospective/retrospective cohorts, case series, case-control, cross sectional, case series, case reports, etc, perhaps similar to epidemiological studies in the list)\n",
    "4. Review (literature reviews, non-systematic). -> We are generating this\n",
    "5. Systematic review and/or meta-analysis\n",
    "6. Non-clinical observational studies (aggregated demographic characteristics, humidity, pollution, population density and temperature) ->won’t be able to classify this with our data or with Kaggle, so please ignore it for now\n",
    "7. Other"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
