{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Thanks to Kriti Mahajan. This table was developed with study type features extracted from the meta data.\n",
    "Code found here: https://colab.research.google.com/drive/1lay-yWJ5ohN94lQLHYIxtWMuDWtCRGy0#scrollTo=N-IhAeEkcxDe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Strategy:\n",
    "1. Combine newest CoY dataset, Kaggle dataset, PubMed dataset and other annotation group's dataset all into one annotation table.\n",
    "2. Feature engineer via revising ngram list/reverse engineer\n",
    "3. If desperate, get synthetic labels via synthetic augmentation of text data (trade out word for synonym)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Trying direct merge with mf and kaggle label set via sha."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>url</th>\n",
       "      <th>Statistical Methods</th>\n",
       "      <th>statistical_method_details</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>study_type_details</th>\n",
       "      <th>results</th>\n",
       "      <th>coronavirus</th>\n",
       "      <th>fatality</th>\n",
       "      <th>sample_size</th>\n",
       "      <th>paper_focus</th>\n",
       "      <th>variables_controlled_for</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>b2897e1277f56641193a6db73825f707eed3e4c9</td>\n",
       "      <td>Sequence requirements for RNA strand transfer ...</td>\n",
       "      <td>Nidovirus subgenomic mRNAs contain a leader se...</td>\n",
       "      <td>2001-12-17</td>\n",
       "      <td>Pasternak, Alexander O.; van den Born, Erwin; ...</td>\n",
       "      <td>http://europepmc.org/articles/pmc125340?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "      <td>Here we present results of a comprehensive co-...</td>\n",
       "      <td>Here we present results of a comprehensive co-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>e3d0d482ebd9a8ba81c254cc433f314142e72174</td>\n",
       "      <td>Crystal structure of murine sCEACAM1a[1,4]: a ...</td>\n",
       "      <td>CEACAM1 is a member of the carcinoembryonic an...</td>\n",
       "      <td>2002-05-01</td>\n",
       "      <td>Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...</td>\n",
       "      <td>http://europepmc.org/articles/pmc125375?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Here we report the crystal structure of solubl...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>00b1d99e70f779eb4ede50059db469c65e8c1469</td>\n",
       "      <td>Synthesis of a novel hepatitis C virus protein...</td>\n",
       "      <td>Hepatitis C virus (HCV) is an important human ...</td>\n",
       "      <td>2001-07-16</td>\n",
       "      <td>Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0</td>\n",
       "      <td>Structure of coronavirus main proteinase revea...</td>\n",
       "      <td>The key enzyme in coronavirus polyprotein proc...</td>\n",
       "      <td>2002-07-01</td>\n",
       "      <td>Anand, Kanchan; Palm, Gottfried J.; Mesters, J...</td>\n",
       "      <td>http://europepmc.org/articles/pmc126080?pdf=re...</td>\n",
       "      <td>['model']</td>\n",
       "      <td>Molecular modelling and mutagenesis data impli...</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "      <td>Molecular modelling and mutagenesis data impli...</td>\n",
       "      <td>The study illustrates that RNA viruses have ev...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>dde02f11923815e6a16a31dd6298c46b109c5dfa</td>\n",
       "      <td>Discontinuous and non-discontinuous subgenomic...</td>\n",
       "      <td>Arteri-, corona-, toro- and roniviruses are ev...</td>\n",
       "      <td>2002-12-01</td>\n",
       "      <td>van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...</td>\n",
       "      <td>http://europepmc.org/articles/pmc136939?pdf=re...</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>We show that BEV mRNAs 3â€“5 lack a leader seque...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59125</th>\n",
       "      <td>59125</td>\n",
       "      <td>pwq5gir9</td>\n",
       "      <td>20f185b11fa23da8938efb5c839c6c078c66f1e9</td>\n",
       "      <td>Cutting-Edge Infectious Disease Diagnostics wi...</td>\n",
       "      <td>Three recent Science articles (Chen et al., 20...</td>\n",
       "      <td>2018-06-13</td>\n",
       "      <td>Chiu, Charles</td>\n",
       "      <td>https://doi.org/10.1016/j.chom.2018.05.016</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>Three recent Science articles (Chen et al., 20...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59126</th>\n",
       "      <td>59126</td>\n",
       "      <td>3racv2qe</td>\n",
       "      <td>e84532333db8f219f20ae0a8fe18130c0d010b04</td>\n",
       "      <td>Intranasal administration of oxytocin: Behavio...</td>\n",
       "      <td>Abstract The intranasal (IN-) administration o...</td>\n",
       "      <td>2013-09-30</td>\n",
       "      <td>Veening, Jan G.; Olivier, Berend</td>\n",
       "      <td>https://doi.org/10.1016/j.neubiorev.2013.04.012</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['review']</td>\n",
       "      <td>The present review explores our current knowle...</td>\n",
       "      <td>Only very recently in a microdialysis study in...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59127</th>\n",
       "      <td>59127</td>\n",
       "      <td>jngto9kv</td>\n",
       "      <td>31105078a2953217223699d09c6a80d0f5edfdf6</td>\n",
       "      <td>Infecciones virales graves en pacientes inmuno...</td>\n",
       "      <td>Resumen Los virus tienen un papel importante d...</td>\n",
       "      <td>2011-04-30</td>\n",
       "      <td>DÃ­az, A.; Zaragoza, R.; Granada, R.; Salavert, M.</td>\n",
       "      <td>https://doi.org/10.1016/j.medin.2011.01.001</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['review']</td>\n",
       "      <td>A review of the literature has been performed ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Furthermore, ventilator-associated pneumonia (...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59128</th>\n",
       "      <td>59128</td>\n",
       "      <td>td0776wj</td>\n",
       "      <td>b1f977d58a0eba25acdee5912ef71c372b0632de</td>\n",
       "      <td>18 Omics, Bioinformatics, and Infectious Disea...</td>\n",
       "      <td>Bioinformatics and the â€˜omicsâ€™ studies are rap...</td>\n",
       "      <td>2011-12-31</td>\n",
       "      <td>Paszkiewicz, Konrad H.; Giezen, Mark van der</td>\n",
       "      <td>https://doi.org/10.1016/b978-0-12-384890-1.000...</td>\n",
       "      <td>['sequencing', 'methods']</td>\n",
       "      <td>In addition, we provide an overview of methods...</td>\n",
       "      <td>['bioinformatics']</td>\n",
       "      <td>Bioinformatics and the â€˜omicsâ€™ studies are rap...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59129</th>\n",
       "      <td>59129</td>\n",
       "      <td>jxbr6hbk</td>\n",
       "      <td>4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...</td>\n",
       "      <td>Identifying spatio-temporal dynamics of Ebola ...</td>\n",
       "      <td>Containing the recent West African outbreak of...</td>\n",
       "      <td>2017-11-29</td>\n",
       "      <td>Gustafson, Kyle B.; Proctor, Joshua L.</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>['model', 'forecast', ' tool']</td>\n",
       "      <td>Despite recent progress in analysing and model...</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>Calibrated to phylogenetic linkages of virus g...</td>\n",
       "      <td>Consistent with other investigations, our resu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59130 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0  paper_id  \\\n",
       "0               0  zjufx4fo   \n",
       "1               1  ymceytj3   \n",
       "2               2  wzj2glte   \n",
       "3               3  2sfqsfm1   \n",
       "4               4  i0zym7iq   \n",
       "...           ...       ...   \n",
       "59125       59125  pwq5gir9   \n",
       "59126       59126  3racv2qe   \n",
       "59127       59127  jngto9kv   \n",
       "59128       59128  td0776wj   \n",
       "59129       59129  jxbr6hbk   \n",
       "\n",
       "                                                     sha  \\\n",
       "0               b2897e1277f56641193a6db73825f707eed3e4c9   \n",
       "1               e3d0d482ebd9a8ba81c254cc433f314142e72174   \n",
       "2               00b1d99e70f779eb4ede50059db469c65e8c1469   \n",
       "3               cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   \n",
       "4               dde02f11923815e6a16a31dd6298c46b109c5dfa   \n",
       "...                                                  ...   \n",
       "59125           20f185b11fa23da8938efb5c839c6c078c66f1e9   \n",
       "59126           e84532333db8f219f20ae0a8fe18130c0d010b04   \n",
       "59127           31105078a2953217223699d09c6a80d0f5edfdf6   \n",
       "59128           b1f977d58a0eba25acdee5912ef71c372b0632de   \n",
       "59129  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   \n",
       "\n",
       "                                                   title  \\\n",
       "0      Sequence requirements for RNA strand transfer ...   \n",
       "1      Crystal structure of murine sCEACAM1a[1,4]: a ...   \n",
       "2      Synthesis of a novel hepatitis C virus protein...   \n",
       "3      Structure of coronavirus main proteinase revea...   \n",
       "4      Discontinuous and non-discontinuous subgenomic...   \n",
       "...                                                  ...   \n",
       "59125  Cutting-Edge Infectious Disease Diagnostics wi...   \n",
       "59126  Intranasal administration of oxytocin: Behavio...   \n",
       "59127  Infecciones virales graves en pacientes inmuno...   \n",
       "59128  18 Omics, Bioinformatics, and Infectious Disea...   \n",
       "59129  Identifying spatio-temporal dynamics of Ebola ...   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "0      Nidovirus subgenomic mRNAs contain a leader se...   2001-12-17   \n",
       "1      CEACAM1 is a member of the carcinoembryonic an...   2002-05-01   \n",
       "2      Hepatitis C virus (HCV) is an important human ...   2001-07-16   \n",
       "3      The key enzyme in coronavirus polyprotein proc...   2002-07-01   \n",
       "4      Arteri-, corona-, toro- and roniviruses are ev...   2002-12-01   \n",
       "...                                                  ...          ...   \n",
       "59125  Three recent Science articles (Chen et al., 20...   2018-06-13   \n",
       "59126  Abstract The intranasal (IN-) administration o...   2013-09-30   \n",
       "59127  Resumen Los virus tienen un papel importante d...   2011-04-30   \n",
       "59128  Bioinformatics and the â€˜omicsâ€™ studies are rap...   2011-12-31   \n",
       "59129  Containing the recent West African outbreak of...   2017-11-29   \n",
       "\n",
       "                                                 authors  \\\n",
       "0      Pasternak, Alexander O.; van den Born, Erwin; ...   \n",
       "1      Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...   \n",
       "2      Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...   \n",
       "3      Anand, Kanchan; Palm, Gottfried J.; Mesters, J...   \n",
       "4      van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...   \n",
       "...                                                  ...   \n",
       "59125                                      Chiu, Charles   \n",
       "59126                   Veening, Jan G.; Olivier, Berend   \n",
       "59127  DÃ­az, A.; Zaragoza, R.; Granada, R.; Salavert, M.   \n",
       "59128       Paszkiewicz, Konrad H.; Giezen, Mark van der   \n",
       "59129             Gustafson, Kyle B.; Proctor, Joshua L.   \n",
       "\n",
       "                                                     url  \\\n",
       "0      http://europepmc.org/articles/pmc125340?pdf=re...   \n",
       "1      http://europepmc.org/articles/pmc125375?pdf=re...   \n",
       "2      https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...   \n",
       "3      http://europepmc.org/articles/pmc126080?pdf=re...   \n",
       "4      http://europepmc.org/articles/pmc136939?pdf=re...   \n",
       "...                                                  ...   \n",
       "59125         https://doi.org/10.1016/j.chom.2018.05.016   \n",
       "59126    https://doi.org/10.1016/j.neubiorev.2013.04.012   \n",
       "59127        https://doi.org/10.1016/j.medin.2011.01.001   \n",
       "59128  https://doi.org/10.1016/b978-0-12-384890-1.000...   \n",
       "59129  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   \n",
       "\n",
       "                  Statistical Methods  \\\n",
       "0                                  []   \n",
       "1                                  []   \n",
       "2                                  []   \n",
       "3                           ['model']   \n",
       "4                                  []   \n",
       "...                               ...   \n",
       "59125                              []   \n",
       "59126                              []   \n",
       "59127                              []   \n",
       "59128       ['sequencing', 'methods']   \n",
       "59129  ['model', 'forecast', ' tool']   \n",
       "\n",
       "                              statistical_method_details          Study Type  \\\n",
       "0                                                    NaN     ['mutagenesis']   \n",
       "1                                                    NaN                  []   \n",
       "2                                                    NaN                  []   \n",
       "3      Molecular modelling and mutagenesis data impli...     ['mutagenesis']   \n",
       "4                                                    NaN                  []   \n",
       "...                                                  ...                 ...   \n",
       "59125                                                NaN        ['clinical']   \n",
       "59126                                                NaN          ['review']   \n",
       "59127                                                NaN          ['review']   \n",
       "59128  In addition, we provide an overview of methods...  ['bioinformatics']   \n",
       "59129  Despite recent progress in analysing and model...         ['genetic']   \n",
       "\n",
       "                                      study_type_details  \\\n",
       "0      Here we present results of a comprehensive co-...   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3      Molecular modelling and mutagenesis data impli...   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "59125  Three recent Science articles (Chen et al., 20...   \n",
       "59126  The present review explores our current knowle...   \n",
       "59127  A review of the literature has been performed ...   \n",
       "59128  Bioinformatics and the â€˜omicsâ€™ studies are rap...   \n",
       "59129  Calibrated to phylogenetic linkages of virus g...   \n",
       "\n",
       "                                                 results coronavirus  \\\n",
       "0      Here we present results of a comprehensive co-...         NaN   \n",
       "1      Here we report the crystal structure of solubl...         NaN   \n",
       "2                                                    NaN         NaN   \n",
       "3      The study illustrates that RNA viruses have ev...         NaN   \n",
       "4      We show that BEV mRNAs 3â€“5 lack a leader seque...         NaN   \n",
       "...                                                  ...         ...   \n",
       "59125                                                NaN         NaN   \n",
       "59126  Only very recently in a microdialysis study in...         NaN   \n",
       "59127                                                NaN         NaN   \n",
       "59128                                                NaN         NaN   \n",
       "59129  Consistent with other investigations, our resu...         NaN   \n",
       "\n",
       "                                                fatality sample_size  \\\n",
       "0                                                    NaN         NaN   \n",
       "1                                                    NaN         NaN   \n",
       "2                                                    NaN         NaN   \n",
       "3                                                    NaN         NaN   \n",
       "4                                                    NaN         NaN   \n",
       "...                                                  ...         ...   \n",
       "59125                                                NaN         NaN   \n",
       "59126                                                NaN         NaN   \n",
       "59127  Furthermore, ventilator-associated pneumonia (...         NaN   \n",
       "59128                                                NaN         NaN   \n",
       "59129                                                NaN         NaN   \n",
       "\n",
       "      paper_focus variables_controlled_for  \n",
       "0             NaN                      NaN  \n",
       "1             NaN                      NaN  \n",
       "2             NaN                      NaN  \n",
       "3             NaN                      NaN  \n",
       "4             NaN                      NaN  \n",
       "...           ...                      ...  \n",
       "59125         NaN                      NaN  \n",
       "59126         NaN                      NaN  \n",
       "59127         NaN                      NaN  \n",
       "59128         NaN                      NaN  \n",
       "59129         NaN                      NaN  \n",
       "\n",
       "[59130 rows x 18 columns]"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf = pd.read_csv('metadata_features-2.csv')\n",
    "mf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Unnamed: 0',\n",
       " 'paper_id',\n",
       " 'sha',\n",
       " 'title',\n",
       " 'abstract',\n",
       " 'publish_time',\n",
       " 'authors',\n",
       " 'url',\n",
       " 'Statistical Methods',\n",
       " 'statistical_method_details',\n",
       " 'Study Type',\n",
       " 'study_type_details',\n",
       " 'results',\n",
       " 'coronavirus',\n",
       " 'fatality',\n",
       " 'sample_size',\n",
       " 'paper_focus',\n",
       " 'variables_controlled_for']"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(mf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>b2897e1277f56641193a6db73825f707eed3e4c9</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ymceytj3</td>\n",
       "      <td>e3d0d482ebd9a8ba81c254cc433f314142e72174</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>wzj2glte</td>\n",
       "      <td>00b1d99e70f779eb4ede50059db469c65e8c1469</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>i0zym7iq</td>\n",
       "      <td>dde02f11923815e6a16a31dd6298c46b109c5dfa</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59125</th>\n",
       "      <td>pwq5gir9</td>\n",
       "      <td>20f185b11fa23da8938efb5c839c6c078c66f1e9</td>\n",
       "      <td>['clinical']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59126</th>\n",
       "      <td>3racv2qe</td>\n",
       "      <td>e84532333db8f219f20ae0a8fe18130c0d010b04</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59127</th>\n",
       "      <td>jngto9kv</td>\n",
       "      <td>31105078a2953217223699d09c6a80d0f5edfdf6</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59128</th>\n",
       "      <td>td0776wj</td>\n",
       "      <td>b1f977d58a0eba25acdee5912ef71c372b0632de</td>\n",
       "      <td>['bioinformatics']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59129</th>\n",
       "      <td>jxbr6hbk</td>\n",
       "      <td>4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59130 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       paper_id                                                sha  \\\n",
       "0      zjufx4fo           b2897e1277f56641193a6db73825f707eed3e4c9   \n",
       "1      ymceytj3           e3d0d482ebd9a8ba81c254cc433f314142e72174   \n",
       "2      wzj2glte           00b1d99e70f779eb4ede50059db469c65e8c1469   \n",
       "3      2sfqsfm1           cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   \n",
       "4      i0zym7iq           dde02f11923815e6a16a31dd6298c46b109c5dfa   \n",
       "...         ...                                                ...   \n",
       "59125  pwq5gir9           20f185b11fa23da8938efb5c839c6c078c66f1e9   \n",
       "59126  3racv2qe           e84532333db8f219f20ae0a8fe18130c0d010b04   \n",
       "59127  jngto9kv           31105078a2953217223699d09c6a80d0f5edfdf6   \n",
       "59128  td0776wj           b1f977d58a0eba25acdee5912ef71c372b0632de   \n",
       "59129  jxbr6hbk  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   \n",
       "\n",
       "               Study Type  \n",
       "0         ['mutagenesis']  \n",
       "1                      []  \n",
       "2                      []  \n",
       "3         ['mutagenesis']  \n",
       "4                      []  \n",
       "...                   ...  \n",
       "59125        ['clinical']  \n",
       "59126          ['review']  \n",
       "59127          ['review']  \n",
       "59128  ['bioinformatics']  \n",
       "59129         ['genetic']  \n",
       "\n",
       "[59130 rows x 3 columns]"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features = mf.drop(columns = ['Unnamed: 0',\n",
    " 'title',\n",
    " 'abstract',\n",
    " 'publish_time',\n",
    " 'authors',\n",
    " 'url',\n",
    " 'Statistical Methods',\n",
    " 'statistical_method_details',\n",
    " 'study_type_details',\n",
    " 'results',\n",
    " 'coronavirus',\n",
    " 'fatality',\n",
    " 'sample_size',\n",
    " 'paper_focus',\n",
    " 'variables_controlled_for'])\n",
    "features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]                                                                           37266\n",
       "['clinical']                                                                  5782\n",
       "['review']                                                                    3594\n",
       "['genetic']                                                                   3089\n",
       "['review', 'clinical']                                                        1347\n",
       "                                                                             ...  \n",
       "['retrospective study', 'retrospectively analyzed']                              1\n",
       "['retrospective study', 'histories']                                             1\n",
       "['review', 'retrospective study', 'retrospectively analyzed', 'clinical']        1\n",
       "['review', 'single center study', 'clinical']                                    1\n",
       "['meta-analysis', 'clinical', 'bioinformatics']                                  1\n",
       "Name: Study Type, Length: 393, dtype: int64"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features['Study Type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>zjufx4fo</td>\n",
       "      <td>b2897e1277f56641193a6db73825f707eed3e4c9</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2sfqsfm1</td>\n",
       "      <td>cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0</td>\n",
       "      <td>['mutagenesis']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>gi6uaa83</td>\n",
       "      <td>8ae137c8da1607b3a8e4c946c07ca8bda67f88ac</td>\n",
       "      <td>['genetic', 'histories']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0m32ecnu</td>\n",
       "      <td>23bc55d6f63fab18b02004483888db2b6a0bfa48</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>fy4w7xz8</td>\n",
       "      <td>0104f6ceccf92ae8567a0102f89cbb976969a774</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59125</th>\n",
       "      <td>pwq5gir9</td>\n",
       "      <td>20f185b11fa23da8938efb5c839c6c078c66f1e9</td>\n",
       "      <td>['clinical']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59126</th>\n",
       "      <td>3racv2qe</td>\n",
       "      <td>e84532333db8f219f20ae0a8fe18130c0d010b04</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59127</th>\n",
       "      <td>jngto9kv</td>\n",
       "      <td>31105078a2953217223699d09c6a80d0f5edfdf6</td>\n",
       "      <td>['review']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59128</th>\n",
       "      <td>td0776wj</td>\n",
       "      <td>b1f977d58a0eba25acdee5912ef71c372b0632de</td>\n",
       "      <td>['bioinformatics']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59129</th>\n",
       "      <td>jxbr6hbk</td>\n",
       "      <td>4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...</td>\n",
       "      <td>['genetic']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21864 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       paper_id                                                sha  \\\n",
       "0      zjufx4fo           b2897e1277f56641193a6db73825f707eed3e4c9   \n",
       "3      2sfqsfm1           cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   \n",
       "6      gi6uaa83           8ae137c8da1607b3a8e4c946c07ca8bda67f88ac   \n",
       "7      0m32ecnu           23bc55d6f63fab18b02004483888db2b6a0bfa48   \n",
       "10     fy4w7xz8           0104f6ceccf92ae8567a0102f89cbb976969a774   \n",
       "...         ...                                                ...   \n",
       "59125  pwq5gir9           20f185b11fa23da8938efb5c839c6c078c66f1e9   \n",
       "59126  3racv2qe           e84532333db8f219f20ae0a8fe18130c0d010b04   \n",
       "59127  jngto9kv           31105078a2953217223699d09c6a80d0f5edfdf6   \n",
       "59128  td0776wj           b1f977d58a0eba25acdee5912ef71c372b0632de   \n",
       "59129  jxbr6hbk  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   \n",
       "\n",
       "                     Study Type  \n",
       "0               ['mutagenesis']  \n",
       "3               ['mutagenesis']  \n",
       "6      ['genetic', 'histories']  \n",
       "7                   ['genetic']  \n",
       "10                  ['genetic']  \n",
       "...                         ...  \n",
       "59125              ['clinical']  \n",
       "59126                ['review']  \n",
       "59127                ['review']  \n",
       "59128        ['bioinformatics']  \n",
       "59129               ['genetic']  \n",
       "\n",
       "[21864 rows x 3 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "to_drop = ['[]']\n",
    "features = features[~features['Study Type'].isin(to_drop)]\n",
    "features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['clinical']                                                        5782\n",
       "['review']                                                          3594\n",
       "['genetic']                                                         3089\n",
       "['review', 'clinical']                                              1347\n",
       "['experimental']                                                    1241\n",
       "                                                                    ... \n",
       "['review', 'retrospective cohort', 'case series', 'clinical']          1\n",
       "['review', 'retrospective study', 'clinical', 'laboratory data']       1\n",
       "['experimental', 'meta-analysis']                                      1\n",
       "['time series', 'time-series', 'clinical']                             1\n",
       "['mutagenesis', 'review', 'life cycle']                                1\n",
       "Name: Study Type, Length: 392, dtype: int64"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features['Study Type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "kaggle_labels = pd.read_csv('design-2.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>8</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnu8gw6w</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>['time series', 'clinical']</td>\n",
       "      <td>5</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ulk3euw</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>['cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>loowpch7</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>['prospective cohort', 'prospective cohort']</td>\n",
       "      <td>4</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>dycx8i7h</td>\n",
       "      <td>2d0d272a793b99ef896547024c51f486e9552e5a</td>\n",
       "      <td>['genetic']</td>\n",
       "      <td>0</td>\n",
       "      <td>2d0d272a793b99ef896547024c51f486e9552e5a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>p8vhi6d9</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>['experimental']</td>\n",
       "      <td>2</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>a6i73v0o</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>i2vg8zqc</td>\n",
       "      <td>e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802</td>\n",
       "      <td>['experimental', 'clinical']</td>\n",
       "      <td>0</td>\n",
       "      <td>e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>nbbcacy4</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>['case study']</td>\n",
       "      <td>8</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>498 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     paper_id                                       sha  \\\n",
       "0    47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1    9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2    hnu8gw6w  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3    5ulk3euw  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4    loowpch7  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "..        ...                                       ...   \n",
       "493  dycx8i7h  2d0d272a793b99ef896547024c51f486e9552e5a   \n",
       "494  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "495  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "496  i2vg8zqc  e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802   \n",
       "497  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                         Study Type  label  \\\n",
       "0                                        ['review']      1   \n",
       "1              ['questionnaire', 'cross-sectional']      8   \n",
       "2                       ['time series', 'clinical']      5   \n",
       "3                               ['cross-sectional']      7   \n",
       "4      ['prospective cohort', 'prospective cohort']      4   \n",
       "..                                              ...    ...   \n",
       "493                                     ['genetic']      0   \n",
       "494                                ['experimental']      2   \n",
       "495  ['questionnaire', 'survey', 'cross-sectional']      7   \n",
       "496                    ['experimental', 'clinical']      0   \n",
       "497                                  ['case study']      8   \n",
       "\n",
       "                                           id  \n",
       "0    6a3fa8ed278df0d05c5e009521de11c72308f60b  \n",
       "1    eb4b8f43f68db5e493984b7e4e56d9f804d7e885  \n",
       "2    2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9  \n",
       "3    9f7ed2037968c63fa97eb6fc7c35e9738a654541  \n",
       "4    de707f97b534ca340482600ed1f008b20d7b1b8e  \n",
       "..                                        ...  \n",
       "493  2d0d272a793b99ef896547024c51f486e9552e5a  \n",
       "494  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb  \n",
       "495  95d9bac431797e2cf8b7294a648ceec3bd50c84b  \n",
       "496  e9dfbc4084497e74dcaf6bd9c1dbde2b0fb70802  \n",
       "497  76edb6d0bd7c47765d3a89032aa610f410506988  \n",
       "\n",
       "[498 rows x 5 columns]"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kg_merge = features.merge(kaggle_labels, how='inner', left_on='sha', right_on='id')\n",
    "kg_merge"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Complete kg_merge annotation set by:\n",
    "1. mapping labels to cw_labels\n",
    "2. performing one-hot-encoding to output multi-label dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [],
   "source": [
    "cw_labels = pd.read_csv('CoY_newlist.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Assignee',\n",
       " 'cord_uid',\n",
       " 'sha',\n",
       " 'title',\n",
       " 'url',\n",
       " 'Computational ',\n",
       " 'Experimental-in vitro',\n",
       " 'Experimental-in vivo',\n",
       " 'Systematic review and/or meta-analysis',\n",
       " 'Clinical-interventional',\n",
       " 'Clinical-interventional.1',\n",
       " 'Clinical-observational',\n",
       " 'Clinical-observational.1',\n",
       " 'Systematic review and/or meta-analysis ',\n",
       " 'Other',\n",
       " 'General Notes']"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(cw_labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [],
   "source": [
    "cw_labels = cw_labels.drop(columns = ['Other', 'General Notes'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Rename/Clean-up labels\n",
    "cw_labels = cw_labels.rename(columns={'Computational ': 'Computational', 'Experimental-in vitro' : 'Experimental - in vitro', 'Experimental-in vivo' : 'Experimental - in vivo', 'Clinical-interventional.1' : 'Clinical-interventional', 'Clinical-observational.1' : 'Clinical-observational', 'Systematic review and/or meta-analysis ' : 'Systematic review and/or meta-analysis'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Assignee',\n",
       " 'cord_uid',\n",
       " 'sha',\n",
       " 'title',\n",
       " 'url',\n",
       " 'Computational',\n",
       " 'Experimental - in vitro',\n",
       " 'Experimental - in vivo',\n",
       " 'Systematic review and/or meta-analysis',\n",
       " 'Clinical-interventional',\n",
       " 'Clinical-interventional',\n",
       " 'Clinical-observational',\n",
       " 'Clinical-observational',\n",
       " 'Systematic review and/or meta-analysis']"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(cw_labels)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MAPPING SCHEME\n",
    "\n",
    "Mapping Kaggle labels to CoronaWhy labels\n",
    "\n",
    "0. Other -> Drop\n",
    "1. Systematic review -> Systematic review and/or meta-analysis\n",
    "2. Randomized control trial -> Clinical-interventional\n",
    "3. Non-randomized trial -> Clinical - interventional\n",
    "4. Prospective observational -> Clinical -observational\n",
    "5. Time-to-event analysis -> Clinical-observational\n",
    "6. Retrospective observational -> Clinical-observational\n",
    "7. Cross-sectional -> Clinical-observational\n",
    "8. Case series -> Clinical-observational\n",
    "9. Modeling -> Computational"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 8, 5, 7, 4, 2, 9, 6, 0, 3])"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kg_merge.explode('label')\n",
    "kg_merge['label'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic_map={\n",
    "         1:'Systematic review and/or meta-analysis',\n",
    "         2:'Clinical-interventional',\n",
    "         3:'Clinical-interventional',\n",
    "         4:'Clinical-observational',\n",
    "         5:'Clinical-observational',\n",
    "         6:'Clinical-observational',\n",
    "         7:'Clinical-observational',\n",
    "         8:'Clinical-observational',\n",
    "         9:'Computational',\n",
    "         0:'Other'\n",
    "        }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "      <th>id</th>\n",
       "      <th>mapped</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>Systematic review and/or meta-analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>8</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnu8gw6w</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>['time series', 'clinical']</td>\n",
       "      <td>5</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ulk3euw</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>['cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>loowpch7</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>['prospective cohort', 'prospective cohort']</td>\n",
       "      <td>4</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   paper_id                                       sha  \\\n",
       "0  47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1  9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2  hnu8gw6w  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3  5ulk3euw  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4  loowpch7  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "\n",
       "                                     Study Type  label  \\\n",
       "0                                    ['review']      1   \n",
       "1          ['questionnaire', 'cross-sectional']      8   \n",
       "2                   ['time series', 'clinical']      5   \n",
       "3                           ['cross-sectional']      7   \n",
       "4  ['prospective cohort', 'prospective cohort']      4   \n",
       "\n",
       "                                         id  \\\n",
       "0  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "\n",
       "                                   mapped  \n",
       "0  Systematic review and/or meta-analysis  \n",
       "1                  Clinical-observational  \n",
       "2                  Clinical-observational  \n",
       "3                  Clinical-observational  \n",
       "4                  Clinical-observational  "
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kg_merge['mapped'] = kg_merge.apply(lambda x: dic_map[x.label], axis = 1) \n",
    "kg_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>sha</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>label</th>\n",
       "      <th>id</th>\n",
       "      <th>mapped</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47ema2dq</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>['review']</td>\n",
       "      <td>1</td>\n",
       "      <td>6a3fa8ed278df0d05c5e009521de11c72308f60b</td>\n",
       "      <td>Systematic review and/or meta-analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9trzqzep</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>['questionnaire', 'cross-sectional']</td>\n",
       "      <td>8</td>\n",
       "      <td>eb4b8f43f68db5e493984b7e4e56d9f804d7e885</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnu8gw6w</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>['time series', 'clinical']</td>\n",
       "      <td>5</td>\n",
       "      <td>2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ulk3euw</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>['cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>9f7ed2037968c63fa97eb6fc7c35e9738a654541</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>loowpch7</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>['prospective cohort', 'prospective cohort']</td>\n",
       "      <td>4</td>\n",
       "      <td>de707f97b534ca340482600ed1f008b20d7b1b8e</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>l4649du3</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>2</td>\n",
       "      <td>bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e</td>\n",
       "      <td>Clinical-interventional</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>3vpoxfnv</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "      <td>['clinical']</td>\n",
       "      <td>9</td>\n",
       "      <td>eb720970f362e4416b73393f230c8606c837c6dc</td>\n",
       "      <td>Computational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>p8vhi6d9</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>['experimental']</td>\n",
       "      <td>2</td>\n",
       "      <td>f778c6317cdc3e2c4aa7a8c7734fa13da10723eb</td>\n",
       "      <td>Clinical-interventional</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>a6i73v0o</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>['questionnaire', 'survey', 'cross-sectional']</td>\n",
       "      <td>7</td>\n",
       "      <td>95d9bac431797e2cf8b7294a648ceec3bd50c84b</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>nbbcacy4</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>['case study']</td>\n",
       "      <td>8</td>\n",
       "      <td>76edb6d0bd7c47765d3a89032aa610f410506988</td>\n",
       "      <td>Clinical-observational</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>398 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     paper_id                                       sha  \\\n",
       "0    47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1    9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2    hnu8gw6w  2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3    5ulk3euw  9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4    loowpch7  de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "..        ...                                       ...   \n",
       "490  l4649du3  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e   \n",
       "491  3vpoxfnv  eb720970f362e4416b73393f230c8606c837c6dc   \n",
       "494  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "495  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "497  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                         Study Type  label  \\\n",
       "0                                        ['review']      1   \n",
       "1              ['questionnaire', 'cross-sectional']      8   \n",
       "2                       ['time series', 'clinical']      5   \n",
       "3                               ['cross-sectional']      7   \n",
       "4      ['prospective cohort', 'prospective cohort']      4   \n",
       "..                                              ...    ...   \n",
       "490                                    ['clinical']      2   \n",
       "491                                    ['clinical']      9   \n",
       "494                                ['experimental']      2   \n",
       "495  ['questionnaire', 'survey', 'cross-sectional']      7   \n",
       "497                                  ['case study']      8   \n",
       "\n",
       "                                           id  \\\n",
       "0    6a3fa8ed278df0d05c5e009521de11c72308f60b   \n",
       "1    eb4b8f43f68db5e493984b7e4e56d9f804d7e885   \n",
       "2    2afcf2ab2330cb56ec493f82d87fa0c5b5f076e9   \n",
       "3    9f7ed2037968c63fa97eb6fc7c35e9738a654541   \n",
       "4    de707f97b534ca340482600ed1f008b20d7b1b8e   \n",
       "..                                        ...   \n",
       "490  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e   \n",
       "491  eb720970f362e4416b73393f230c8606c837c6dc   \n",
       "494  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   \n",
       "495  95d9bac431797e2cf8b7294a648ceec3bd50c84b   \n",
       "497  76edb6d0bd7c47765d3a89032aa610f410506988   \n",
       "\n",
       "                                     mapped  \n",
       "0    Systematic review and/or meta-analysis  \n",
       "1                    Clinical-observational  \n",
       "2                    Clinical-observational  \n",
       "3                    Clinical-observational  \n",
       "4                    Clinical-observational  \n",
       "..                                      ...  \n",
       "490                 Clinical-interventional  \n",
       "491                           Computational  \n",
       "494                 Clinical-interventional  \n",
       "495                  Clinical-observational  \n",
       "497                  Clinical-observational  \n",
       "\n",
       "[398 rows x 6 columns]"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Drop 0. Other label\n",
    "kg_merge = kg_merge[kg_merge.label != 0]\n",
    "kg_merge"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "100 less annotations is a huge drop. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### One-Hot Encoding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
